From a27e1e1596c06d82bafe62637f4155661ca9c6c6 Mon Sep 17 00:00:00 2001
From: Gymxo <55960119+Gymxo@users.noreply.github.com>
Date: Sun, 28 Mar 2021 10:54:31 -0400
Subject: [PATCH 2/2] Update vncsession.c

---
 unix/vncserver/vncsession.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/unix/vncserver/vncsession.c b/unix/vncserver/vncsession.c
index 784d5dae..b9a5a80f 100644
--- a/unix/vncserver/vncsession.c
+++ b/unix/vncserver/vncsession.c
@@ -37,6 +37,8 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 
+#define HOST_NAME_MAX _POSIX_HOST_NAME_MAX
+
 extern char **environ;
 
 // PAM service name
-- 
2.24.3 (Apple Git-128)

