From e8de29f895bebe5005c603ea2ff40db816da94d5 Mon Sep 17 00:00:00 2001
From: Hayden Seay <pinseeker75@outlook.com>
Date: Sat, 17 Jul 2021 20:13:40 +0000
Subject: [PATCH] [add-shell] Remove unnecessary awk call

---
 add-shell | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/add-shell b/add-shell
index abce1c1..8d08ec2 100755
--- a/add-shell
+++ b/add-shell
@@ -17,7 +17,7 @@ cleanup() {
 }
 trap cleanup EXIT
 
-if ! awk '{print}' "$file" > "$tmpfile"
+if ! cat "$file" > "$tmpfile"
 then
         cat 1>&2 <<EOF
 Either another instance of $0 is running, or it was previously interrupted.
-- 
GitLab

