diff -urN ruby/coroutine/ucontext/Context.h build_work/iphoneos-arm64/1600/ruby/coroutine/ucontext/Context.h
--- ruby/coroutine/ucontext/Context.h	2020-12-24 22:33:01.000000000 -0500
+++ build_work/iphoneos-arm64/1600/ruby/coroutine/ucontext/Context.h	2021-01-02 18:08:12.000000000 -0500
@@ -9,7 +9,17 @@
 
 #include <assert.h>
 #include <stddef.h>
+
+#if 1
+#include <libucontext.h>
+#define ucontext_t libucontext_ucontext_t
+#define getcontext libucontext_getcontext
+#define setcontext libucontext_setcontext
+#define swapcontext libucontext_swapcontext
+#define makecontext libucontext_makecontext
+#else
 #include <ucontext.h>
+#endif
 
 #define COROUTINE __attribute__((noreturn)) void
 
