diff -urN dpkg-1.20.0/data/cputable dpkg/data/cputable
--- dpkg-1.20.0/data/cputable	2019-11-05 06:59:03.000000000 -0500
+++ dpkg/data/cputable	2020-06-09 01:42:14.325741148 -0400
@@ -21,6 +21,7 @@
 alpha		alpha		alpha.*			64	little
 amd64		x86_64		(amd64|x86_64)		64	little
 armeb		armeb		arm.*b			32	big
+armk		armk		arm.*k			32	little
 arm		arm		arm.*			32	little
 arm64		aarch64		aarch64			64	little
 avr32		avr32		avr32			32	big
diff -urN dpkg-1.20.0/lib/dpkg/command.c dpkg/lib/dpkg/command.c
--- dpkg-1.20.0/lib/dpkg/command.c	2020-01-17 18:07:01.000000000 -0500
+++ dpkg/lib/dpkg/command.c	2020-06-09 01:42:14.325741148 -0400
@@ -18,6 +18,15 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
+/* wrapper functions to make iOS shell scripts function correctly */
+#if defined(__APPLE__)
+#  include <TargetConditionals.h>
+#  if TARGET_OS_IPHONE
+#    define ENABLE_IOSEXEC
+#    include <libiosexec.h>
+#  endif
+#endif
+
 #include <config.h>
 #include <compat.h>
 
diff -urN dpkg-1.20.0/src/help.c dpkg/src/help.c
--- dpkg-1.20.0/src/help.c	2020-01-17 18:07:01.000000000 -0500
+++ dpkg/src/help.c	2020-06-09 01:42:14.321741090 -0400
@@ -121,7 +121,6 @@
     /* Mac OS X uses dyld (Mach-O) instead of ld.so (ELF), and does not have
      * an ldconfig. */
 #if defined(__APPLE__) && defined(__MACH__)
-    "update_dyld_shared_cache",
 #elif defined(__GLIBC__) || defined(__UCLIBC__) || \
       defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
     "ldconfig",
