Build fails with abs
diff -ruN node-v14.13.0/deps/v8/src/objects/bigint.cc nodejs/tools/deps/v8/src/objects/bigint.cc
--- node-v14.13.0/deps/v8/src/objects/bigint.cc	2020-09-30 01:58:15.000000000 +0900
+++ nodejs/tools/deps/v8/src/objects/bigint.cc	2020-10-07 11:00:38.000000000 +0900
@@ -863,7 +865,7 @@
     STATIC_ASSERT(sizeof(digit_t) >= sizeof(value));
     return (x->length() == 1) && (x->sign() == (value < 0)) &&
            (x->digit(0) ==
-            static_cast<digit_t>(std::abs(static_cast<int64_t>(value))));
+            static_cast<digit_t>(std::fabs(static_cast<int64_t>(value))));
   }
   DCHECK(y->IsHeapNumber());
   double value = Handle<HeapNumber>::cast(y)->value();
@@ -887,7 +889,7 @@
     STATIC_ASSERT(sizeof(digit_t) >= sizeof(y_value));
     if (x->length() > 1) return AbsoluteGreater(x_sign);
 
-    digit_t abs_value = std::abs(static_cast<int64_t>(y_value));
+    digit_t abs_value = std::fabs(static_cast<int64_t>(y_value));
     digit_t x_digit = x->digit(0);
     if (x_digit > abs_value) return AbsoluteGreater(x_sign);
     if (x_digit < abs_value) return AbsoluteLess(x_sign);
