diff -ruN node-v14.13.0/deps/uv/src/unix/darwin.c nodejs/deps/uv/src/unix/darwin.c
--- node-v14.13.0/deps/uv/src/unix/darwin.c	2020-09-30 01:58:13.000000000 +0900
+++ nodejs/deps/uv/src/unix/darwin.c	2020-10-07 10:52:44.000000000 +0900
@@ -33,9 +33,7 @@
 #include <sys/sysctl.h>
 #include <unistd.h>  /* sysconf */
 
-#if !TARGET_OS_IPHONE
 #include "darwin-stub.h"
-#endif
 
 static uv_once_t once = UV_ONCE_INIT;
 static uint64_t (*time_func)(void);