--- Makefile.old        2020-07-19 18:08:41.598000000 +0800
+++ Makefile    2020-07-19 18:08:47.738794986 +0800
@@ -61,27 +61,27 @@
 
 LDLIBS=libgif.a -lm
 
-all: libgif.so libgif.a libutil.so libutil.a $(UTILS)
+all: libgif.dylib libgif.a libutil.dylib libutil.a $(UTILS)
        $(MAKE) -C doc
 
 $(UTILS):: libgif.a libutil.a
 
-libgif.so: $(OBJECTS) $(HEADERS)
-       $(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
+libgif.dylib: $(OBJECTS) $(HEADERS)
+       $(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(OBJECTS) -o libgif.dylib
 
 libgif.a: $(OBJECTS) $(HEADERS)
        $(AR) rcs libgif.a $(OBJECTS)
 
-libutil.so: $(UOBJECTS) $(UHEADERS)
-       $(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libutil.so.$(LIBMAJOR) -o libutil.so $(UOBJECTS)
+libutil.dylib: $(UOBJECTS) $(UHEADERS)
+       $(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(OBJECTS) -o libutil.dylib $(UOBJECTS)
 
 libutil.a: $(UOBJECTS) $(UHEADERS)
        $(AR) rcs libutil.a $(UOBJECTS)
 
 clean:
-       rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a libgif.so libutil.a libutil.so *.o
-       rm -f libgif.so.$(LIBMAJOR).$(LIBMINOR).$(LIBPOINT)
-       rm -f libgif.so.$(LIBMAJOR)
+       rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a libgif.dylib libutil.a libutil.dylib *.o
+       rm -f libgif.dylib.$(LIBMAJOR).$(LIBMINOR).$(LIBPOINT)
+       rm -f libgif.dylib.$(LIBMAJOR)
        rm -fr doc/*.1 *.html doc/staging
 
 check: all
@@ -99,9 +99,9 @@
 install-lib:
        $(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
        $(INSTALL) -m 644 libgif.a "$(DESTDIR)$(LIBDIR)/libgif.a"
-       $(INSTALL) -m 755 libgif.so "$(DESTDIR)$(LIBDIR)/libgif.so.$(LIBVER)"
-       ln -sf libgif.so.$(LIBVER) "$(DESTDIR)$(LIBDIR)/libgif.so.$(LIBMAJOR)"
-       ln -sf libgif.so.$(LIBMAJOR) "$(DESTDIR)$(LIBDIR)/libgif.so"
+       $(INSTALL) -m 755 libgif.dylib "$(DESTDIR)$(LIBDIR)/libgif.dylib.$(LIBVER)"
+       ln -sf libgif.dylib.$(LIBVER) "$(DESTDIR)$(LIBDIR)/libgif.dylib.$(LIBMAJOR)"
+       ln -sf libgif.dylib.$(LIBMAJOR) "$(DESTDIR)$(LIBDIR)/libgif.dylib"
 install-man:
        $(INSTALL) -d "$(DESTDIR)$(MANDIR)/man1"
        $(INSTALL) -m 644 doc/*.1 "$(DESTDIR)$(MANDIR)/man1"
@@ -112,7 +112,7 @@
        rm -f "$(DESTDIR)$(INCDIR)/gif_lib.h"
 uninstall-lib:
        cd "$(DESTDIR)$(LIBDIR)" && \
-               rm -f libgif.a libgif.so libgif.so.$(LIBMAJOR) libgif.so.$(LIBVER)
+               rm -f libgif.a libgif.dylib libgif.dylib.$(LIBMAJOR) libgif.dylib.$(LIBVER)
 uninstall-man:
        cd "$(DESTDIR)$(MANDIR)/man1" && rm -f $(shell cd doc >/dev/null && echo *.1)
 