--- a/src/libkrw.c	2024-04-20 14:46:12.796173476 -0700
+++ b/src/libkrw.c	2024-04-20 18:01:14.500961452 -0700
@@ -160,3 +160,8 @@
     if (krw_handlers.physwrite == NULL) return ENOTSUP;
     return krw_handlers.physwrite(from, to, len, granule);
 }
+
+__attribute__((visibility("hidden")))
+int libkrw_initialization(krw_handlers_t handlers) {
+  return EPROTONOSUPPORT;
+}

--- a/src/libkrw_tfp0.c	2021-11-01 15:47:05.000000000 -0700
+++ b/src/libkrw_tfp0.c	2024-04-20 17:59:39.170186064 -0700
@@ -204,8 +204,8 @@
     return EDEVERR;
 }

-__attribute__((visibility("hidden")))
-int libkrw_initialization(krw_handlers_t handlers) {
+__attribute__((used))
+int krw_initializer(krw_handlers_t handlers) {
     // Make sure structure version is not lower than what we compiled with
     if (handlers->version < LIBKRW_HANDLERS_VERSION) {
         return EPROTONOSUPPORT;
