diff -urN build_work/iphoneos-arm64/1500/berkeleydb/dist/Makefile.in build_work/iphoneos-arm64/1600/berkeleydb/dist/Makefile.in
--- berkeleydb/dist/Makefile.in	2020-05-29 13:58:22.000000000 -0400
+++ berkeleydb/dist/Makefile.in	2020-08-05 21:55:22.000000000 -0400
@@ -1215,7 +1215,7 @@
 # Library and standard utilities install.
 ##################################################
 library_install: install_setup
-library_install: install_include install_lib install_utilities @ADDITIONAL_INSTALL@ install_docs
+library_install: install_include install_lib install_utilities @ADDITIONAL_INSTALL@
 
 uninstall: uninstall_include uninstall_lib uninstall_utilities @ADDITIONAL_UNINSTALL@ uninstall_docs
 