From: Tobias Frost <tobi@coldtobi.de>>
Subject: Disable UBC feature on hppa since it causes test suite failures.
Updateig patch for tokyocabinet 1.47
Index: tokyocabinet-1.4.47/myconf.h
===================================================================
--- tokyocabinet-1.4.47.orig/myconf.h	2011-02-11 16:37:44.000000000 +0100
+++ tokyocabinet-1.4.47/myconf.h	2012-03-05 21:18:59.660815523 +0100
@@ -153,7 +153,7 @@
 #define TCITOHLL(TC_num)  (TC_num)
 #endif
 
-#if defined(_MYNOUBC)
+#if defined(_MYNOUBC) || defined(__hppa__)
 #define TCUBCACHE      0
 #elif defined(_SYS_LINUX_) || defined(_SYS_FREEBSD_) || defined(_SYS_NETBSD_) || \
   defined(_SYS_MACOSX_) || defined(_SYS_SUNOS_)
