Description: Makefile change
Authors: NIIBE Yutaka <gniibe@fsij.org>
Reviewed-By: NIIBE Yutaka <gniibe@fsij.org>
Last-Update: 2010-04-27

--- ttyrec-1.0.8.orig/Makefile	2006-06-12 00:52:50.000000000 +0900
+++ ttyrec/Makefile	2010-04-27 09:58:23.000000000 +0900
@@ -1,7 +1,10 @@
 CC = gcc
-CFLAGS = -O2
+DEFS= -DSVR4 -D_GNU_SOURCE
+CFLAGS = -O2 -g -fomit-frame-pointer -Wall $(DEFS)
 VERSION = 1.0.8
 
+DESTDIR=
+
 TARGET = ttyrec ttyplay ttytime
 
 DIST =	ttyrec.c ttyplay.c ttyrec.h io.c io.h ttytime.c\
@@ -18,6 +21,9 @@
 ttytime: ttytime.o io.o
 	$(CC) $(CFLAGS) -o ttytime ttytime.o io.o
 
+install:
+	install ttyrec ttyplay ttytime $(DESTDIR)/usr/bin
+
 clean:
 	rm -f *.o $(TARGET) ttyrecord *~
 
