diff -ruN libsoundio-2.0.0/CMakeLists.txt libsoundio/CMakeLists.txt
--- libsoundio-2.0.0/CMakeLists.txt	2019-02-20 21:25:44.000000000 +0800
+++ libsoundio/CMakeLists.txt	2021-03-31 08:18:26.089114628 +0800
@@ -107,7 +107,7 @@
         include_directories(${COREFOUNDATION_INCLUDE_DIR})
 
         find_path(AUDIOUNIT_INCLUDE_DIR NAMES AudioUnit.h)
-        find_library(AUDIOUNIT_LIBRARY NAMES AudioUnit)
+        find_library(AUDIOUNIT_LIBRARY NAMES AudioToolbox)
         include_directories(${AUDIOUNIT_INCLUDE_DIR})
     else()
         set(STATUS_COREAUDIO "not found")
