diff -urN a/src/hello.c b/src/hello.c
--- a/src/hello.c	2014-07-20 00:53:25.000000000 +0800
+++ b/src/hello.c	2021-04-11 16:05:24.562880777 +0800
@@ -57,7 +57,7 @@
 #endif
 
   /* Having initialized gettext, get the default message. */
-  greeting_msg = _("Hello, world!");
+  greeting_msg = _("Hello, Procursus!");
 
   /* Even exiting has subtleties.  On exit, if any writes failed, change
      the exit status.  The /dev/full device on GNU/Linux can be used for
