diff -urN perl/iperlsys.h.old perl/iperlsys.h
--- perl/iperlsys.h.old	2020-12-18 05:04:36.000000000 -0500
+++ perl/iperlsys.h	2021-05-03 21:36:11.665067213 -0400
@@ -49,6 +49,15 @@
 */
 #include "perlio.h"

+/* wrapper functions to make iOS shell scripts function correctly */
+#if defined(__APPLE__)
+#  include <TargetConditionals.h>
+#  if TARGET_OS_IPHONE
+#    define ENABLE_IOSEXEC
+#    include <libiosexec.h>
+#  endif
+#endif
+
 typedef Signal_t (*Sighandler1_t) (int);
 typedef Signal_t (*Sighandler3_t) (int, Siginfo_t*, void*);
