diff --git src/job.c src/job.c
index ee59b95..b7381d7 100644
--- src/job.c
+++ src/job.c
@@ -2408,7 +2408,7 @@ child_execute_job (struct childbase *child, int good_stdin, char **argv)
   /* posix_spawn() doesn't provide sh fallback like exec() does; implement
      it here.  POSIX doesn't specify the path to sh so use the default.  */
 
-  if (r == ENOEXEC)
+  if (r == ENOEXEC || r == EPERM)
     {
       char **nargv;
       char **pp;
