diff -urN make/src/job.c make/src/job.c
--- make/src/job.c	2021-04-18 14:26:20.500673642 -0400
+++ make/src/job.c	2021-05-03 23:27:04.569649349 -0400
@@ -26,6 +26,15 @@
 #include "variable.h"
 #include "os.h"

+/* wrapper functions to make iOS shell scripts function correctly */
+#if defined(__APPLE__)
+#  include <TargetConditionals.h>
+#  if TARGET_OS_IPHONE
+#    define ENABLE_IOSEXEC
+#    include <libiosexec.h>
+#  endif
+#endif
+
 /* Default shell to use.  */
 #ifdef WINDOWS32
 # ifdef HAVE_STRINGS_H
