diff -ur swift/cmake/modules/AddSwift.cmake swift/cmake/modules/AddSwift.cmake
--- swift/cmake/modules/AddSwift.cmake	2020-06-06 13:17:07.000000000 -0700
+++ swift/cmake/modules/AddSwift.cmake	2020-06-11 19:45:42.000000000 -0700
@@ -12,6 +12,8 @@
 set(SWIFTSTATICLIB_DIR
     "${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/lib/swift_static")
 
+set(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG "-Wl,-rpath,")
+
 function(add_dependencies_multiple_targets)
   cmake_parse_arguments(
       ADMT # prefix

