diff --git ld64/src/ld/Options.cpp ld64/src/ld/Options.cpp
index 3bb8324..6b5b6c8 100644
--- ld64/src/ld/Options.cpp
+++ ld64/src/ld/Options.cpp
@@ -4436,7 +4436,7 @@ void Options::buildSearchPaths(int argc, const char* argv[])
 					}
 					else {
 						fLibrarySearchPaths.push_back(strdup(newPath));
-						sdkOverride = true;
+						//sdkOverride = true;
 					}
 				}
 			}
@@ -5783,7 +5783,7 @@ void Options::reconfigureDefaults()
 	}
 
 	// <rdar://problem/51911409> codesign all userland arm64 macOS binaries
-	if ( dyldLoadsOutput() && (fArchitecture == CPU_TYPE_ARM64) && platforms().contains(ld::Platform::macOS) )
+	if ( dyldLoadsOutput() && (fArchitecture == CPU_TYPE_ARM64) )
 		fAdHocSign = true;
 
 }
