diff --color -urN mtree-netbsd-debian-20180822-6/compare.c mtree-netbsd/compare.c
--- mtree-netbsd-debian-20180822-6/compare.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/compare.c	2021-03-31 14:24:13.620831383 -0400
@@ -36,7 +36,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
@@ -89,11 +88,15 @@
 #ifndef NO_SHA2
 #if HAVE_SHA2_H && HAVE_SHA512_FILE
 #include <sha2.h>
+#include <sha256.h>
+#include <sha512.h>
 #else
 #include <nbcompat/sha2.h>
 #endif
 #endif
 
+#include <sys/stat.h>
+#include <sys/time.h>
 
 #include "extern.h"
 
diff --color -urN mtree-netbsd-debian-20180822-6/crc.c mtree-netbsd/crc.c
--- mtree-netbsd-debian-20180822-6/crc.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/crc.c	2021-03-31 14:22:50.892998740 -0400
@@ -39,7 +39,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/create.c mtree-netbsd/create.c
--- mtree-netbsd-debian-20180822-6/create.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/create.c	2021-03-31 14:23:24.416929025 -0400
@@ -36,7 +36,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
@@ -110,6 +109,8 @@
 #ifndef NO_SHA2
 #if HAVE_SHA2_H && HAVE_SHA512_FILE
 #include <sha2.h>
+#include <sha256.h>
+#include <sha512.h>
 #else
 #include <nbcompat/sha2.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/excludes.c mtree-netbsd/excludes.c
--- mtree-netbsd-debian-20180822-6/excludes.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/excludes.c	2021-03-31 14:22:50.892998740 -0400
@@ -36,7 +36,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/extern.h mtree-netbsd/extern.h
--- mtree-netbsd-debian-20180822-6/extern.h	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/extern.h	2021-03-31 14:22:50.892998740 -0400
@@ -44,7 +44,6 @@
 #endif
 #endif
  
-#include <nbcompat.h>
 #if HAVE_ERR_H
 #include <err.h> 
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/getid.c mtree-netbsd/getid.c
--- mtree-netbsd-debian-20180822-6/getid.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/getid.c	2021-03-31 14:22:50.892998740 -0400
@@ -74,7 +74,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_PARAM_H
 #include <sys/param.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/misc.c mtree-netbsd/misc.c
--- mtree-netbsd-debian-20180822-6/misc.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/misc.c	2021-03-31 14:22:50.892998740 -0400
@@ -38,7 +38,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/mtree.c mtree-netbsd/mtree.c
--- mtree-netbsd-debian-20180822-6/mtree.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/mtree.c	2021-03-31 14:22:50.892998740 -0400
@@ -36,7 +36,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/pack_dev.c mtree-netbsd/pack_dev.c
--- mtree-netbsd-debian-20180822-6/pack_dev.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/pack_dev.c	2021-03-31 14:23:47.628882290 -0400
@@ -43,7 +43,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
@@ -59,7 +58,7 @@
 #endif
 
 /* Linux libc has these here */
-#include <sys/sysmacros.h>
+#include <sys/types.h>
 
 #if HAVE_LIMITS_H
 #include <limits.h>
diff --color -urN mtree-netbsd-debian-20180822-6/spec.c mtree-netbsd/spec.c
--- mtree-netbsd-debian-20180822-6/spec.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/spec.c	2021-03-31 14:22:50.892998740 -0400
@@ -72,7 +72,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/stat_flags.c mtree-netbsd/stat_flags.c
--- mtree-netbsd-debian-20180822-6/stat_flags.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/stat_flags.c	2021-03-31 14:22:50.892998740 -0400
@@ -40,7 +40,6 @@
 #endif
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
diff --color -urN mtree-netbsd-debian-20180822-6/verify.c mtree-netbsd/verify.c
--- mtree-netbsd-debian-20180822-6/verify.c	2021-01-12 10:49:06.000000000 -0500
+++ mtree-netbsd/verify.c	2021-03-31 14:22:50.892998740 -0400
@@ -36,7 +36,6 @@
 #include "nbtool_config.h"
 #endif
 
-#include <nbcompat.h>
 #if HAVE_SYS_CDEFS_H
 #include <sys/cdefs.h>
 #endif
