diff -urN mosh-1.3.2/Makefile.am mosh/Makefile.am
--- mosh-1.3.2/Makefile.am	2017-07-22 21:14:53.000000000 +0000
+++ mosh/Makefile.am	2021-08-28 02:16:28.000000000 +0000
@@ -5,29 +5,29 @@
 CLANG_SCAN_BUILD = scan-build
 AM_DISTCHECK_CONFIGURE_FLAGS = --enable-compile-warnings=distcheck --enable-examples
 
-.PHONY:	VERSION
+.PHONY:	VERSION.txt
 
-VERSION:
+VERSION.txt:
 	@echo @PACKAGE_STRING@ > VERSION.dist
 	@set -e; if git describe --dirty --always > VERSION.git 2>&1 && \
 		[ -z `git rev-parse --show-prefix` ]; then \
 		if ! diff -q VERSION.git VERSION > /dev/null 2>&1; then \
-			mv -f VERSION.git VERSION; \
+			mv -f VERSION.git VERSION.txt; \
 		fi; \
-	elif ! diff -q VERSION.dist VERSION > /dev/null 2>&1; then \
-		mv -f VERSION.dist VERSION; \
+	elif ! diff -q VERSION.dist VERSION.txt > /dev/null 2>&1; then \
+		mv -f VERSION.dist VERSION.txt; \
 	fi
 	@rm -f VERSION.dist VERSION.git
 
-version.h:	VERSION
-	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION)" > version.h.new
+version.h:	VERSION.txt
+	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION.txt)" > version.h.new
 	@set -e; if ! diff -q version.h version.h.new > /dev/null 2>&1; then \
 		mv -f version.h.new version.h; \
 	fi
 	@rm -f version.h.new
 
 clean-local:
-	@rm -rf version.h VERSION cov-int mosh-coverity.txz
+	@rm -rf version.h VERSION.txt cov-int mosh-coverity.txz
 
 cppcheck: $(BUILT_SOURCES) config.h
 	cppcheck --enable=all --template=gcc -include config.h -I . \
diff -urN mosh-1.3.2/scripts/Makefile.am mosh/scripts/Makefile.am
--- mosh-1.3.2/scripts/Makefile.am	2017-07-22 21:14:53.000000000 +0000
+++ mosh/scripts/Makefile.am	2021-08-28 02:17:01.000000000 +0000
@@ -4,6 +4,6 @@
 endif
 CLEANFILES = $(bin_SCRIPTS)
 
-mosh:	mosh.pl ../VERSION Makefile
-	@sed -e "s/\@VERSION\@/`cat ../VERSION`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
+mosh:	mosh.pl ../VERSION.txt Makefile
+	@sed -e "s/\@VERSION\@/`cat ../VERSION.txt`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
 	@chmod a+x mosh
