--- a/src/faketime.c	2022-03-05 03:28:35.000000000 +0800
+++ b/src/faketime.c	2022-03-13 08:37:38.582667238 +0800
@@ -50,11 +50,19 @@
 
 const char version[] = "0.9.10";
 
+/*
+ * If somehow someday Procursus decided to use
+ * the BSD(Apple)'s date(1), remember to change
+ * this part to correctly use coreutils' date
+ */
+/*
 #if (defined __APPLE__) || (defined __sun)
 static const char *date_cmd = "gdate";
 #else
 static const char *date_cmd = "date";
 #endif
+*/
+static const char *date_cmd = "date";
 
 #define PATH_BUFSIZE 4096
 
@@ -263,7 +271,7 @@
     if (SEM_FAILED == (sem = sem_open(sem_name, O_CREAT|O_EXCL, S_IWUSR|S_IRUSR, 1)))
     {
       perror("faketime: sem_open");
-      fprintf(stderr, "The faketime wrapper only works on platforms that support the sem_open()\nsystem call. However, you may LD_PRELOAD libfaketime without using this wrapper.\n");
+      fprintf(stderr, "The faketime wrapper only works on platforms that support the sem_open()\nsystem call. However, you may %s libfaketime without using this wrapper.\n", __APPLE__ ? "DYLD_INSERT_LIBRARIES" : "LD_PRELOAD");
       exit(EXIT_FAILURE);
     }
 
