diff -urN a/lib/dpkg/dpkg.h b/lib/dpkg/dpkg.h
--- a/lib/dpkg/dpkg.h
+++ b/lib/dpkg/dpkg.h
@@ -68,6 +68,7 @@
 #define CONTROLFILE        "control"
 #define CONFFILESFILE      "conffiles"
 #define PREINSTFILE        "preinst"
+#define EXTRAINSTFILE      "extrainst_"
 #define POSTINSTFILE       "postinst"
 #define PRERMFILE          "prerm"
 #define POSTRMFILE         "postrm"
diff -urN a/src/main/unpack.c b/src/main/unpack.c
--- a/src/main/unpack.c
+++ b/src/main/unpack.c
@@ -1577,6 +1577,15 @@
   subproc_reap(pid, BACKEND " --fsys-tarfile", SUBPROC_NOPIPE);

   tar_deferred_extract(newfiles_queue.head, pkg);
+  if (oldversionstatus == PKG_STAT_NOTINSTALLED || oldversionstatus == PKG_STAT_CONFIGFILES) {
+    maintscript_new(pkg, EXTRAINSTFILE, "extra-installation", cidir, cidirrest,
+                    "install", NULL);
+  } else {
+    maintscript_new(pkg, EXTRAINSTFILE, "extra-installation", cidir, cidirrest,
+                    "upgrade",
+                    versiondescribe(&pkg->installed.version, vdew_nonambig),
+                    NULL);
+  }

   if (oldversionstatus == PKG_STAT_HALFINSTALLED ||
       oldversionstatus == PKG_STAT_UNPACKED) {
