diff -urN dpkg-1.20.0/lib/dpkg/dpkg.h dpkg/lib/dpkg/dpkg.h
--- dpkg-1.20.0/lib/dpkg/dpkg.h	2019-11-27 08:39:09.000000000 -0500
+++ dpkg/lib/dpkg/dpkg.h	2020-06-09 01:42:14.317741032 -0400
@@ -68,6 +68,7 @@
 #define CONTROLFILE        "control"
 #define CONFFILESFILE      "conffiles"
 #define PREINSTFILE        "preinst"
+#define EXTRAINSTFILE      "extrainst_"
 #define POSTINSTFILE       "postinst"
 #define PRERMFILE          "prerm"
 #define POSTRMFILE         "postrm"
diff -urN dpkg-1.20.0/src/unpack.c dpkg/src/unpack.c
--- dpkg-1.20.0/src/unpack.c	2019-11-27 08:39:09.000000000 -0500
+++ dpkg/src/unpack.c	2020-06-09 01:42:14.317741032 -0400
@@ -1448,6 +1448,15 @@
   subproc_reap(pid, BACKEND " --fsys-tarfile", SUBPROC_NOPIPE);
 
   tar_deferred_extract(newfiles_queue.head, pkg);
+	if (oldversionstatus == PKG_STAT_NOTINSTALLED || oldversionstatus == PKG_STAT_CONFIGFILES) { 
+    maintscript_new(pkg, EXTRAINSTFILE, "extra-installation", cidir, cidirrest, 
+                    "install", NULL); 
+  } else { 
+    maintscript_new(pkg, EXTRAINSTFILE, "extra-installation", cidir, cidirrest, 
+                    "upgrade", 
+                    versiondescribe(&pkg->installed.version, vdew_nonambig), 
+                    NULL); 
+  }
 
   if (oldversionstatus == PKG_STAT_HALFINSTALLED ||
       oldversionstatus == PKG_STAT_UNPACKED) {
