diff -urN dpkg-1.20.0/src/help.c dpkg/src/help.c
--- dpkg-1.20.0/src/main/help.c	2020-01-17 18:07:01.000000000 -0500
+++ dpkg/src/main/help.c	2020-06-09 01:42:14.321741090 -0400
@@ -121,7 +121,6 @@
     /* Mac OS X uses dyld (Mach-O) instead of ld.so (ELF), and does not have
      * an ldconfig. */
 #if defined(__APPLE__) && defined(__MACH__)
-    "update_dyld_shared_cache",
 #elif defined(__GLIBC__) || defined(__UCLIBC__) || \
       defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
     "ldconfig",
