diff --git a/src/main/help.c b/src/main/help.c
index 56b05ae..c0c19f6 100644
--- a/src/main/help.c
+++ b/src/main/help.c
@@ -92,7 +92,6 @@ checkpath(void)
 		/* macOS uses dyld (Mach-O) instead of ld.so (ELF), and does
 		 * not have an ldconfig. */
 #if defined(__APPLE__) && defined(__MACH__)
-		"update_dyld_shared_cache",
 #elif defined(__GLIBC__) || defined(__UCLIBC__) || \
       defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
 		"ldconfig",
