diff -urN a/src/main/help.c b/src/main/help.c
--- a/src/main/help.c
+++ b/src/main/help.c
@@ -124,7 +124,6 @@
     /* Mac OS X uses dyld (Mach-O) instead of ld.so (ELF), and does not have
      * an ldconfig. */
 #if defined(__APPLE__) && defined(__MACH__)
-    "update_dyld_shared_cache",
 #elif defined(__GLIBC__) || defined(__UCLIBC__) || \
       defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
     "ldconfig",
