From 27ae4e7c2e02e6e9a0d4a584868475c0d9b763b9 Mon Sep 17 00:00:00 2001
From: Noortheen Raja <jnoortheen@gmail.com>
Date: Fri, 21 May 2021 09:44:12 +0530
Subject: [PATCH] fix: skip xonsh.built_ins from amalgam

fixes #4289
---
 xonsh/__init__.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/xonsh/__init__.py b/xonsh/__init__.py
index 22f3149bf4..cc834b12ee 100644
--- a/xonsh/__init__.py
+++ b/xonsh/__init__.py
@@ -3,7 +3,7 @@
 
 # amalgamate exclude jupyter_kernel parser_table completion_parser_table parser_test_table pyghooks
 # amalgamate exclude winutils wizard pytest_plugin fs macutils pygments_cache
-# amalgamate exclude jupyter_shell proc
+# amalgamate exclude jupyter_shell proc built_ins
 import os as _os
 
 if _os.getenv("XONSH_DEBUG", ""):
@@ -86,8 +86,6 @@
         _sys.modules["xonsh.readline_shell"] = __amalgam__
         tracer = __amalgam__
         _sys.modules["xonsh.tracer"] = __amalgam__
-        built_ins = __amalgam__
-        _sys.modules["xonsh.built_ins"] = __amalgam__
         dumb_shell = __amalgam__
         _sys.modules["xonsh.dumb_shell"] = __amalgam__
         execer = __amalgam__
