diff -urN fakeroot/doc/Makefile.am.orig fakeroot/doc/Makefile.am
--- fakeroot/doc/Makefile.am.orig	2020-10-08 13:13:18.000000000 -0400
+++ fakeroot/doc/Makefile.am	2020-11-22 10:15:48.000000000 -0500
@@ -11,17 +11,17 @@
 	rm -rf `find $(distdir) -name .arch-ids`
 	cd $(distdir) && po4a -k 0 --rm-backups --variable "srcdir=$(realpath $(distdir))/" po4a/po4a.cfg
 
-#all-recursive: po4a
+all-recursive: po4a
 
-#po4a:
-#	@$(NORMAL_INSTALL)
-#	po4a -k 0 --variable "srcdir=$(srcdir)/" $(srcdir)/po4a/po4a.cfg
-#
-#clean-recursive: po4a-clean
-#
-#po4a-clean:
-#	@$(NORMAL_INSTALL)
-#	po4a -k 0 --variable "srcdir=$(srcdir)/" \
-#	   --rm-translations $(srcdir)/po4a/po4a.cfg
-#
-#.PHONY: po4a po4a-clean
+po4a:
+	@$(NORMAL_INSTALL)
+	po4a -k 0 --variable "srcdir=$(srcdir)/" $(srcdir)/po4a/po4a.cfg
+
+clean-recursive: po4a-clean
+
+po4a-clean:
+	@$(NORMAL_INSTALL)
+	po4a -k 0 --variable "srcdir=$(srcdir)/" \
+	   --rm-translations $(srcdir)/po4a/po4a.cfg
+
+.PHONY: po4a po4a-clean
