From 68a8f611e01c1e43a7362f590316268ccf7865c5 Mon Sep 17 00:00:00 2001
From: David CARLIER <devnexen@gmail.com>
Date: Wed, 31 Mar 2021 21:28:58 +0100
Subject: [PATCH] coroutine: Darwin on ARM64 needs alignment of 2^2

---
 coroutine/arm64/Context.S | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/coroutine/arm64/Context.S b/coroutine/arm64/Context.S
index 4d371c3dda09..07d50d30df39 100644
--- a/coroutine/arm64/Context.S
+++ b/coroutine/arm64/Context.S
@@ -11,10 +11,12 @@
 #if defined(__APPLE__)
 #define x29 fp
 #define x30 lr
-#endif
-
+.text
+.p2align 2
+#else
 .text
 .align 2
+#endif
 
 .global PREFIXED_SYMBOL(SYMBOL_PREFIX,coroutine_transfer)
 PREFIXED_SYMBOL(SYMBOL_PREFIX,coroutine_transfer):
