From 816a1d97fdcb46cec51f6fa25d191630698106d1 Mon Sep 17 00:00:00 2001
From: David CARLIER <devnexen@gmail.com>
Date: Fri, 5 Mar 2021 20:47:19 +0000
Subject: [PATCH] coroutine mac m1 update. using proper link register and frame
 pointer which equal x30/x29.

---
 coroutine/arm64/Context.S | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/coroutine/arm64/Context.S b/coroutine/arm64/Context.S
index e2bd5b3090b8..4d371c3dda09 100644
--- a/coroutine/arm64/Context.S
+++ b/coroutine/arm64/Context.S
@@ -8,6 +8,11 @@
 #define TOKEN_PASTE(x,y) x##y
 #define PREFIXED_SYMBOL(prefix,name) TOKEN_PASTE(prefix,name)
 
+#if defined(__APPLE__)
+#define x29 fp
+#define x30 lr
+#endif
+
 .text
 .align 2
 
