From d58df830d0823cff346531091d09ee9ed05d786d Mon Sep 17 00:00:00 2001
From: Nick Chan <towinchenmi@gmail.com>
Date: Thu, 19 Dec 2024 16:18:54 +0800
Subject: [PATCH 1/2] fix ios shutdown

Signed-off-by: Nick Chan <towinchenmi@gmail.com>
---
 shutdown/shutdown.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/shutdown/shutdown.c b/shutdown/shutdown.c
index beeee91..97d6496 100644
--- a/shutdown/shutdown.c
+++ b/shutdown/shutdown.c
@@ -71,8 +71,11 @@ __FBSDID("$FreeBSD$");
 #include <vproc.h>
 #include <vproc_priv.h>
 
+#if !(TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR)
 #include "kextmanager.h"
 #include <IOKit/kext/kextmanager_types.h>
+#endif
+
 #include <IOKit/pwr_mgt/IOPMLib.h>
 #include <mach/mach_port.h>		// allocate
 #include <mach/mach.h>			// task_self, etc
@@ -136,11 +139,12 @@ static void nolog(void);
 static void timeout(int);
 static void timewarn(time_t);
 static void usage(const char *);
-#ifdef __APPLE__
 static int audit_shutdown(int);
+#if defined(__APPLE__) && !(TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR)
 static int reserve_reboot(void);
 #endif
 
+int reboot3(int);
 extern const char **environ;
 
 int
@@ -454,7 +458,7 @@ log_and_exec_reboot_or_halt(void)
 die_you_gravy_sucking_pig_dog(void)
 #endif
 {
-#ifndef __APPLE__
+#if defined(__APPLE__) && (TARGET_OS_IPHONE && !TARGET_OS_SIMULTOR)
 	char *empty_environ[] = { NULL };
 #else
 	if ((errno = reserve_reboot())) {
@@ -809,6 +813,8 @@ kextdDisabled(void)
 /*
  * contact kextd to lock for reboot
  */
+
+#if defined(__APPLE__) && !(TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR)
 int
 reserve_reboot(void)
 {
@@ -865,4 +871,5 @@ finish:
 
     return rval;
 }
+#endif
 #endif /* __APPLE__ */
-- 
2.39.1

