diff -urN libgcrypt/random/rndlinux.c.orig libgcrypt/random/rndlinux.c
--- libgcrypt/random/rndlinux.c.orig	2021-04-26 23:45:33.840687376 -0400
+++ libgcrypt/random/rndlinux.c	2021-04-26 23:45:50.096024773 -0400
@@ -36,12 +36,10 @@
 #include <Availability.h>
 #ifdef __MAC_10_11
 #include <TargetConditionals.h>
-#if !defined(TARGET_OS_IPHONE) || TARGET_OS_IPHONE == 0
 extern int getentropy (void *buf, size_t buflen) __attribute__ ((weak_import));
 #define HAVE_GETENTROPY
 #endif
 #endif
-#endif
 #if defined(__linux__) || !defined(HAVE_GETENTROPY)
 #ifdef HAVE_SYSCALL
 # include <sys/syscall.h>
