diff -urN Libnotify-279.40.4/notifyutil/notifyutil.c Libnotify/notifyutil/notifyutil.c
--- Libnotify-279.40.4/notifyutil/notifyutil.c	2019-10-31 18:20:16.000000000 +0000
+++ Libnotify/notifyutil/notifyutil.c	2021-11-30 13:37:50.000000000 +0000
@@ -30,10 +30,9 @@
 #include <sys/time.h>
 #include <mach/mach.h>
 #include <notify.h>
-#include <notify_private.h>
+#include "../notify_private.h"
 #include <signal.h>
 #include <dispatch/dispatch.h>
-#include <os/variant_private.h>
 
 #define forever for(;;)
 #define IndexNull ((uint32_t)-1)
@@ -114,10 +113,7 @@
 	fprintf(stderr, "    -g key         get state value for key\n");
 	fprintf(stderr, "    -s key val     set state value for key\n");
 
-	if(os_variant_has_internal_diagnostics(NULL))
-	{
-		fprintf(stderr, "    --dump         dumps metadata to a file in /var/run/\n");
-	}
+	fprintf(stderr, "    --dump         dumps metadata to a file in /var/run/\n");
 }
 
 // Triggers a notifyd dump
@@ -626,7 +622,7 @@
 				fprintf(stderr, "value following -s name must be a 64-bit integer\n");
 			}
 		}
-		else if (!strcmp(argv[i], "--dump") && os_variant_has_internal_diagnostics(NULL))
+		else if (!strcmp(argv[i], "--dump"))
 		{
 			notifyutil_dump();
 			exit(0);
