diff -urN a/src/main.rs b/src/main.rs
--- a/src/main.rs	2020-12-31 00:00:00.000000000 +0100
+++ b/src/main.rs	2020-12-31 00:00:00.000000000 +0100
@@ -9,7 +9,7 @@
     mpsc::{Receiver, Sender, SyncSender},
   },
 };
-use termion::raw::IntoRawMode;
+use termion::{raw::IntoRawMode, screen::AlternateScreen};
 
 mod compositor;
 pub mod firework;
@@ -45,7 +45,7 @@
   let (regen_buffer_filler, regen_buffer): (SyncSender<Drawables>, Receiver<Drawables>) =
     mpsc::sync_channel(5);
 
-  let mut stdout = stdout()
+  let mut stdout = AlternateScreen::from(stdout())
     .into_raw_mode()
     .expect("Unable to capture stdout. Exiting.");
 
