diff -urN a/Makefile.osx b/Makefile.osx
--- a/Makefile.osx	2020-12-01 22:00:00.000000000 +0100
+++ b/Makefile.osx	2020-12-02 21:30:00.000000000 +0100
@@ -1,22 +1,17 @@
 
-PREFIX		:= /usr/local
-CXX		:= clang++ -stdlib=libc++
-CC		:= clang
-ARCHFLAGS	:= -arch x86_64 -mmacosx-version-min=10.7
-OPTFLAGS	:= -DNDEBUG -ffast-math -mfpmath=sse -msse -msse2 -O3 -ftree-vectorize
+OPTFLAGS	:= -DNDEBUG -ffast-math -freciprocal-math -ftree-vectorize
 
-CXXFLAGS	:= $(ARCHFLAGS) $(OPTFLAGS) -I. -Isrc -Irubberband -I/usr/local/include -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -DHAVE_VDSP -DUSE_SPEEX -DNO_THREAD_CHECKS -DNO_TIMING
+CXXFLAGS	:= $(ARG_CXXFLAGS) $(OPTFLAGS) -I. -Isrc -Irubberband -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -DHAVE_VDSP -DUSE_SPEEX -DNO_THREAD_CHECKS -DNO_TIMING
 
-CFLAGS		:= $(ARCHFLAGS) $(OPTFLAGS)
-LDFLAGS		:= $(ARCHFLAGS) -lpthread $(LDFLAGS)
+CFLAGS		:= $(ARG_CFLAGS) $(OPTFLAGS)
+LDFLAGS		:= -lpthread -lsamplerate -lsndfile $(ARG_LDFLAGS)
 
 LIBRARY_LIBS		:= -framework Accelerate
-PROGRAM_LIBS		:= -L$(PREFIX)/lib -lsndfile $(LIBRARY_LIBS)
-VAMP_PLUGIN_LIBS	:= -L$(PREFIX)/lib -lvamp-sdk $(LIBRARY_LIBS)
+PROGRAM_LIBS		:= -lsndfile $(LIBRARY_LIBS)
+VAMP_PLUGIN_LIBS	:= -lvamp-sdk $(LIBRARY_LIBS)
 LADSPA_PLUGIN_LIBS	:= $(LIBRARY_LIBS)
 
 MKDIR			:= mkdir
-AR			:= ar
 
 INSTALL_BINDIR		:= $(PREFIX)/bin
 INSTALL_INCDIR		:= $(PREFIX)/include/rubberband
@@ -153,13 +148,10 @@
 lib:
 	$(MKDIR) $@
 
-install:	default
+install:
 	$(MKDIR) -p $(DESTDIR)$(INSTALL_BINDIR)
 	$(MKDIR) -p $(DESTDIR)$(INSTALL_INCDIR)
 	$(MKDIR) -p $(DESTDIR)$(INSTALL_LIBDIR)
-	$(MKDIR) -p $(DESTDIR)$(INSTALL_VAMPDIR)
-	$(MKDIR) -p $(DESTDIR)$(INSTALL_LADSPADIR)
-	$(MKDIR) -p $(DESTDIR)$(INSTALL_LRDFDIR)
 	$(MKDIR) -p $(DESTDIR)$(INSTALL_PKGDIR)
 	cp $(PROGRAM_TARGET) $(DESTDIR)$(INSTALL_BINDIR)
 	cp $(PUBLIC_INCLUDES) $(DESTDIR)$(INSTALL_INCDIR)
@@ -169,11 +161,6 @@
 	cp $(DYNAMIC_TARGET) $(DESTDIR)$(INSTALL_LIBDIR)/$(LIBNAME).$(DYNAMIC_FULL_VERSION)$(DYNAMIC_EXTENSION)
 	ln -s $(LIBNAME).$(DYNAMIC_FULL_VERSION)$(DYNAMIC_EXTENSION) $(DESTDIR)$(INSTALL_LIBDIR)/$(LIBNAME).$(DYNAMIC_ABI_VERSION)$(DYNAMIC_EXTENSION)
 	ln -s $(LIBNAME).$(DYNAMIC_FULL_VERSION)$(DYNAMIC_EXTENSION) $(DESTDIR)$(INSTALL_LIBDIR)/$(LIBNAME)$(DYNAMIC_EXTENSION)
-	cp $(VAMP_TARGET) $(DESTDIR)$(INSTALL_VAMPDIR)
-	cp vamp/vamp-rubberband.cat $(DESTDIR)$(INSTALL_VAMPDIR)
-	cp $(LADSPA_TARGET) $(DESTDIR)$(INSTALL_LADSPADIR)
-	cp ladspa/ladspa-rubberband.cat $(DESTDIR)$(INSTALL_LADSPADIR)
-	cp ladspa/ladspa-rubberband.rdf $(DESTDIR)$(INSTALL_LRDFDIR)
 	sed "s,%PREFIX%,$(PREFIX)," rubberband.pc.in \
 	  > $(DESTDIR)$(INSTALL_PKGDIR)/rubberband.pc
 
