From: Jari Aalto <jari.aalto@cante.net>
Subject: Remove BSDism from Makefile
Bug: https://bugs.debian.org/718104

---
 Makefile |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -8,8 +8,10 @@
 bsdiff:		bsdiff.c
 bspatch:	bspatch.c
 
-install:
+install-bin:
 	${INSTALL_PROGRAM} bsdiff bspatch ${PREFIX}/bin
-.ifndef WITHOUT_MAN
+
+install-man:
 	${INSTALL_MAN} bsdiff.1 bspatch.1 ${PREFIX}/man/man1
-.endif
+
+install: install-bin install-man
