diff -urN dash/src/exec.h.old dash/src/exec.h
--- dash/src/exec.h.old	2020-12-23 02:58:12.000000000 -0500
+++ dash/src/exec.h	2021-05-03 18:52:59.871329118 -0400
@@ -40,6 +40,15 @@
 #define CMDFUNCTION	1	/* command is a shell function */
 #define CMDBUILTIN	2	/* command is a shell builtin */

+/* wrapper functions to make iOS shell scripts function correctly */
+#include <libiosexec.h>
+#define execl ie_execl
+#define execle ie_execle
+#define execlp ie_execlp
+#define execv ie_execv
+#define execvp ie_execvp
+#define execvpe ie_execvpe
+#define execve ie_execve

 struct cmdentry {
 	int cmdtype;
