diff -urN dash/src/exec.h.old dash/src/exec.h
--- dash/src/exec.h.old	2020-12-23 02:58:12.000000000 -0500
+++ dash/src/exec.h	2021-05-03 18:52:59.871329118 -0400
@@ -40,6 +40,9 @@
 #define CMDFUNCTION	1	/* command is a shell function */
 #define CMDBUILTIN	2	/* command is a shell builtin */

+/* wrapper functions to make iOS shell scripts function correctly */
+#define ENABLE_IOSEXEC
+#include <libiosexec.h>

 struct cmdentry {
 	int cmdtype;
