From df3f19da716c427e86b329475fed8dcc0fd16e67 Mon Sep 17 00:00:00 2001
From: Nick Chan <towinchenmi@gmail.com>
Date: Thu, 28 Sep 2023 14:47:06 +0800
Subject: [PATCH] membership functions does not work on iOS

---
 newgrp.tproj/newgrp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/newgrp.tproj/newgrp.c b/newgrp.tproj/newgrp.c
index 3a4f412..1512749 100644
--- a/newgrp.tproj/newgrp.c
+++ b/newgrp.tproj/newgrp.c
@@ -158,7 +158,7 @@ addgroup(const char *grpname)
 	gid_t egid;
 	struct group *grp;
 	char *ep, *pass;
-#ifndef __APPLE__
+#if 1
 	char **p;
 #endif
 	char *grp_passwd;
@@ -178,7 +178,7 @@ addgroup(const char *grpname)
 			return;
 		}
 
-#ifdef __APPLE__
+#if 0
 	status = mbr_uid_to_uuid(pwd->pw_uid, user_uuid);
 	if (status)
 		errc(1, status, "mbr_uid_to_uuid");
-- 
2.39.1

