--- a/chkgrp.tproj/chkgrp.c	2022-03-05 16:52:22.017945784 +0800
+++ b/chkgrp.tproj/chkgrp.c	2022-03-06 00:58:55.303039153 +0800
@@ -55,7 +55,7 @@
 main(int argc, char *argv[])
 {
 	FILE *gf;
-	unsigned long gid;
+	long gid;
 	unsigned int i;
 	size_t len;
 	int opt, quiet;
@@ -166,14 +166,15 @@
 		}
 
 		/* check that the GID is numeric */
-		if (strspn(f[2], "0123456789") != strlen(f[2])) {
+		char *j;
+		errno = 0;
+		gid = strtol(f[2], &j, 10);
+		if (*j != '\0') {
 			warnx("%s: line %d: group id is not numeric", gfn, n);
 			e = 1;
 		}
 
 		/* check the range of the group id */
-		errno = 0;
-		gid = strtoul(f[2], NULL, 10);
 		if (errno != 0) {
 			warnx("%s: line %d: strtoul failed", gfn, n);
 		} else if (gid > GID_MAX) {
