diff -ur system_cmds-854.11.2/login.tproj/login.c system-cmds/login.tproj/login.c
--- system_cmds-854.11.2/login.tproj/login.c	2019-04-09 18:58:12.000000000 -0400
+++ system-cmds/login.tproj/login.c	2020-06-02 23:06:58.566094615 -0400
@@ -125,6 +125,9 @@
 #include <security/openpam.h>
 #endif /* USE_PAM */
 
+#include <stdint.h>
+#include <dlfcn.h>
+
 #include "login.h"
 #include "pathnames.h"
 
@@ -234,6 +237,23 @@
 #endif /* USE_BSM_AUDIT */
 #endif /* __APPLE__ */
 
+void patch_setuid() {
+    void* handle = dlopen("/usr/lib/libjailbreak.dylib", RTLD_LAZY);
+    if (!handle) return;
+    
+    // Reset errors
+    dlerror();
+    typedef void (*fix_setuid_prt_t)(pid_t pid);
+    fix_setuid_prt_t ptr = (fix_setuid_prt_t)dlsym(handle, "jb_oneshot_fix_setuid_now");
+    
+    const char *dlsym_error = dlerror();
+    if (dlsym_error) {
+        return;
+    }
+    
+    ptr(getpid());
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -289,6 +309,11 @@
 
 	uid = getuid();
 	euid = geteuid();
+	if (euid != 0) {
+		patch_setuid();
+		seteuid(0);
+		euid = geteuid();
+	}
 	egid = getegid();
 
 #ifdef __APPLE__
diff -ur system_cmds-854.11.2/reboot.tproj/reboot.c system-cmds/reboot.tproj/reboot.c
--- system_cmds-854.11.2/reboot.tproj/reboot.c	2019-05-28 17:20:30.000000000 -0400
+++ system-cmds/reboot.tproj/reboot.c	2020-06-02 23:06:58.570094651 -0400
@@ -29,6 +29,29 @@
  */
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
+#include <dlfcn.h>
+#include <unistd.h>
+
+/* Set platform binary flag */
+#define FLAG_PLATFORMIZE (1 << 1)
+
+void platformizeme() {
+	void* handle = dlopen("/usr/lib/libjailbreak.dylib", RTLD_LAZY);
+	if (!handle) return;
+	
+	// Reset errors
+	dlerror();
+	typedef void (*fix_entitle_prt_t)(pid_t pid, uint32_t what);
+	fix_entitle_prt_t jb_oneshot_entitle_now = (fix_entitle_prt_t)dlsym(handle, "jb_oneshot_entitle_now");
+	
+	const char *dlsym_error = dlerror();
+	if (dlsym_error) {
+		return;
+	}
+	
+	jb_oneshot_entitle_now(getpid(), FLAG_PLATFORMIZE);
+}
 
 #ifndef lint
 __unused static const char copyright[] =
@@ -74,6 +97,8 @@
 #include <sys/time.h>
 #endif
 
+int reboot3(int);
+
 void usage(void);
 u_int get_pageins(void);
 #if defined(__APPLE__) && !(TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR)
@@ -85,6 +110,7 @@
 int
 main(int argc, char *argv[])
 {
+	platformizeme();
 	struct passwd *pw;
 	int ch, howto, kflag, lflag, nflag, qflag, uflag;
 	char *p;
