diff -urN yt-dlp-2023.03.04/Makefile yt-dlp/Makefile
--- yt-dlp-2023.03.04/Makefile	2023-03-04 22:24:51.000000000 +0000
+++ yt-dlp/Makefile	2023-06-20 19:12:33.343555815 +0000
@@ -44,11 +44,9 @@
 # set markdown input format to "markdown-smart" for pandoc version 2 and to "markdown" for pandoc prior to version 2
 MARKDOWN = $(shell if [ `pandoc -v | head -n1 | cut -d" " -f2 | head -c1` = "2" ]; then echo markdown-smart; else echo markdown; fi)
 
-install: lazy-extractors yt-dlp yt-dlp.1 completions
+install: lazy-extractors yt-dlp completions
 	mkdir -p $(DESTDIR)$(BINDIR)
 	install -m755 yt-dlp $(DESTDIR)$(BINDIR)/yt-dlp
-	mkdir -p $(DESTDIR)$(MANDIR)/man1
-	install -m644 yt-dlp.1 $(DESTDIR)$(MANDIR)/man1/yt-dlp.1
 	mkdir -p $(DESTDIR)$(SHAREDIR)/bash-completion/completions
 	install -m644 completions/bash/yt-dlp $(DESTDIR)$(SHAREDIR)/bash-completion/completions/yt-dlp
 	mkdir -p $(DESTDIR)$(SHAREDIR)/zsh/site-functions
