diff --color -urN yt-dlp-2021.09.25/README.md yt-dlp/README.md
--- yt-dlp-2021.09.25/README.md	2021-09-24 20:59:59.000000000 -0400
+++ yt-dlp/README.md	2021-10-04 00:02:08.903398168 -0400
@@ -240,9 +240,6 @@
 ## General Options:
     -h, --help                       Print this help text and exit
     --version                        Print program version and exit
-    -U, --update                     Update this program to latest version. Make
-                                     sure that you have sufficient permissions
-                                     (run with sudo if needed)
     -i, --ignore-errors              Ignore download and postprocessing errors.
                                      The download will be considered successfull
                                      even if the postprocessing fails
diff --color -urN yt-dlp-2021.09.25/yt_dlp/__init__.py yt-dlp/yt_dlp/__init__.py
--- yt-dlp-2021.09.25/yt_dlp/__init__.py	2021-09-24 20:59:59.000000000 -0400
+++ yt-dlp/yt_dlp/__init__.py	2021-10-04 00:04:58.188099262 -0400
@@ -40,7 +40,6 @@
     std_headers,
     write_string,
 )
-from .update import run_update
 from .downloader import (
     FileDownloader,
 )
@@ -744,11 +743,12 @@
 
         # Update version
         if opts.update_self:
-            # If updater returns True, exit. Required for windows
-            if run_update(ydl):
-                if actual_use:
-                    sys.exit('ERROR: The program must exit for the update to complete')
-                sys.exit()
+            parser.error(
+                "yt-dlp's self-update mechanism is disabled on Procursus.\n"
+                "Please update yt-dlp using apt(8).\n"
+                "See https://github.com/ProcursusTeam/Procursus/blob/main/yt-dlp.mk for the "
+                "latest packaged version.\n"
+            )
 
         # Maybe do nothing
         if not actual_use:
diff --color -urN yt-dlp-2021.09.25/yt_dlp/options.py yt-dlp/yt_dlp/options.py
--- yt-dlp-2021.09.25/yt_dlp/options.py	2021-09-24 20:59:59.000000000 -0400
+++ yt-dlp/yt_dlp/options.py	2021-10-04 00:05:59.567598023 -0400
@@ -204,7 +204,7 @@
     general.add_option(
         '-U', '--update',
         action='store_true', dest='update_self',
-        help='Update this program to latest version. Make sure that you have sufficient permissions (run with sudo if needed)')
+        help=optparse.SUPPRESS_HELP)
     general.add_option(
         '-i', '--ignore-errors',
         action='store_true', dest='ignoreerrors',
