diff -urN a/doc/sample.nanorc.in b/doc/sample.nanorc.in
--- a/doc/sample.nanorc.in	2024-09-05 17:29:01.000000000 +0800
+++ b/doc/sample.nanorc.in	2024-12-20 16:02:55.118501838 +0800
@@ -73,7 +73,7 @@
 # set guidestripe 75
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Display a "scrollbar" on the righthand side of the edit window.
 # set indicator
@@ -87,10 +87,10 @@
 ## Enable vim-style lock-files.  This is just to let a vim user know you
 ## are editing a file [s]he is trying to edit and vice versa.  There are
 ## no plans to implement vim-style undo state in these files.
-# set locking
+set locking
 
 ## Fall back to slow libmagic to try and determine an applicable syntax.
-# set magic
+set magic
 
 ## The opening and closing brackets that are found by a matching-bracket
 ## search.  This may not contain blank characters.  The opening brackets
@@ -178,7 +178,7 @@
 
 ## Use the end of the title bar for some state flags: I = auto-indenting,
 ## M = mark, L = hard-wrapping long lines, R = recording, S = soft-wrapping.
-# set stateflags
+set stateflags
 
 ## Use this tab size instead of the default; it must be greater than 0.
 # set tabsize 8
@@ -249,7 +249,7 @@
 ## For all details, see 'man nanorc', section SYNTAX HIGHLIGHTING.
 
 ## To include most of the existing syntax definitions, you can do:
-# include @PKGDATADIR@/*.nanorc
+include @PKGDATADIR@/*.nanorc
 
 ## Or you can select just the ones you need.  For example:
 # include @PKGDATADIR@/html.nanorc
