diff -urN neofetch-7.1.0/neofetch neofetch/neofetch
--- neofetch-7.1.0/neofetch	2020-08-02 11:37:41.000000000 -0700
+++ neofetch/neofetch	2021-03-15 18:12:29.000000000 -0700
@@ -1243,9 +1243,11 @@
                 iPad6,1[12]):        "iPad 5" ;;
                 iPad7,[5-6]):        "iPad 6" ;;
                 iPad7,1[12]):        "iPad 7" ;;
+                iPad11,[67]):        "iPad 8" ;;
                 iPad4,[1-3]):        "iPad Air" ;;
                 iPad5,[3-4]):        "iPad Air 2" ;;
                 iPad11,[3-4]):       "iPad Air 3" ;;
+                iPad13,[1-2]):       "iPad Air 4";;
                 iPad6,[7-8]):        "iPad Pro (12.9 Inch)" ;;
                 iPad6,[3-4]):        "iPad Pro (9.7 Inch)" ;;
                 iPad7,[1-2]):        "iPad Pro 2 (12.9 Inch)" ;;
@@ -1285,6 +1287,10 @@
                 iPhone12,3):    "iPhone 11 Pro" ;;
                 iPhone12,5):    "iPhone 11 Pro Max" ;;
                 iPhone12,8):    "iPhone SE 2020" ;;
+                iPhone13,1):    "iPhone 12 Mini" ;;
+                iPhone13,2):    "iPhone 12" ;;
+                iPhone13,3):    "iPhone 12 Pro" ;;
+                iPhone13,4):    "iPhone 12 Pro Max" ;;
 
                 iPod1,1): "iPod touch" ;;
                 ipod2,1): "iPod touch 2G" ;;
@@ -1292,6 +1298,7 @@
                 ipod4,1): "iPod touch 4G" ;;
                 ipod5,1): "iPod touch 5G" ;;
                 ipod7,1): "iPod touch 6G" ;;
+                iPod9,1): "iPod touch 7G" ;;
             esac
 
             model=$_
@@ -2152,8 +2159,9 @@
                     "Apple A10 Fusion (4) @ 2.34GHz"
                 ;;
                 "iPhone10,"[1-6]): "Apple A11 Bionic (6) @ 2.39GHz" ;;
-                "iPhone11,"[2468] | "iPad11,"[1-4]): "Apple A12 Bionic (6) @ 2.49GHz" ;;
+                "iPhone11,"[2468] | "iPad11,"[1-4] | "iPad11,"[6-7]): "Apple A12 Bionic (6) @ 2.49GHz" ;;
                 "iPhone12,"[1358]): "Apple A13 Bionic (6) @ 2.65GHz" ;;
+                "iPhone13,"[1-4] | "iPad13,"[1-2]): "Apple A14 Bionic (6) @ 3.00Ghz" ;;
 
                 "iPod2,1"): "Samsung S5L8720 (1) @ 533MHz" ;;
                 "iPod3,1"): "Samsung S5L8922 (1) @ 600MHz" ;;
@@ -2483,8 +2491,9 @@
                 "iPhone8,"[1-4] | "iPad6,1"[12]):             "PowerVR GT7600" ;;
                 "iPhone9,"[1-4] | "iPad7,"[5-6]):             "PowerVR GT7600 Plus" ;;
                 "iPhone10,"[1-6]):                            "Apple Designed GPU (A11)" ;;
-                "iPhone11,"[2468]):                           "Apple Designed GPU (A12)" ;;
+                "iPhone11,"[2468] | "iPad11,"[67]):          "Apple Designed GPU (A12)" ;;
                 "iPhone12,"[1358]):                           "Apple Designed GPU (A13)" ;;
+                "iPhone13,"[1234] | "iPad13,"[12]):           "Apple Designed GPU (A14)" ;;
 
                 "iPad3,"[1-3]):     "PowerVR SGX534MP4" ;;
                 "iPad3,"[4-6]):     "PowerVR SGX554MP4" ;;
