diff -urN neofetch-7.1.0/neofetch neofetch/neofetch
--- neofetch-7.1.0/neofetch	2020-08-02 11:37:41.000000000 -0700
+++ neofetch/neofetch	2021-05-13 22:28:15.000000000 -0700
@@ -1571,6 +1571,7 @@
             has port  && tot port installed && ((packages-=1))
             has brew  && dir /usr/local/Cellar/*
             has pkgin && tot pkgin list
+            has dpkg  && tot dpkg-query -f '.\n' -W
 
             has nix-store && {
                 manager=nix-system && tot nix-store -q --requisites "/run/current-system/sw"
