diff -urN tcc/tccrun.c.orig tcc/tccrun.c
--- tcc/tccrun.c.orig	2023-02-22 21:47:29
+++ tcc/tccrun.c	2023-02-22 21:56:38
@@ -468,8 +468,14 @@
 /* XXX: BSD sometimes dump core with bad system call */
 # if (defined TCC_TARGET_ARM && !TARGETOS_BSD) || defined TCC_TARGET_ARM64
     if (mode == 0 || mode == 3) {
+/* XXX: On Darwin systems, __clear_cache just calls sys_icache_invalidate */
+#if defined(__APPLE__)
+        #include <libkern/OSCacheControl.h>
+        sys_icache_invalidate(ptr, length);
+#else
         void __clear_cache(void *beginning, void *end);
         __clear_cache(ptr, (char *)ptr + length);
+#endif
     }
 # endif

