diff --git a/configure.local b/configure.local
new file mode 100644
index 00000000..5a6f3891
--- /dev/null
+++ b/configure.local
@@ -0,0 +1,53 @@
+HAVE_FTS=1
+HAVE_FTS_COMPARE_CONST=0
+HAVE_GETLINE=1
+HAVE_GETSUBOPT=1
+HAVE_ISBLANK=1
+HAVE_LESS_T=1
+HAVE_MKDTEMP=1
+HAVE_NTOHL=1
+HAVE_O_DIRECTORY=1
+HAVE_OHASH=0
+HAVE_PATH_MAX=1
+HAVE_PLEDGE=0
+HAVE_PROGNAME=1
+HAVE_REALLOCARRAY=0
+HAVE_RECALLOCARRAY=0
+HAVE_REWB_BSD=1
+HAVE_REWB_SYSV=0
+HAVE_STRCASESTR=1
+HAVE_STRINGLIST=1
+HAVE_STRLCAT=1
+HAVE_STRLCPY=1
+HAVE_STRPTIME=1
+HAVE_STRSEP=1
+HAVE_STRTONUM=0 # Was added in iOS 14
+HAVE_ENDIAN=0
+HAVE_SYS_ENDIAN=0
+HAVE_VASPRINTF=1
+HAVE_WCHAR=1
+HAVE_NANOSLEEP=1
+LD_NANOSLEEP=""
+HAVE_RECVMSG=1
+HAVE_CMSG=1
+MANPATH_DEFAULT="/usr/share/man:/usr/local/share/man:/usr/local/man:@MEMO_PREFIX@@MEMO_SUB_PREFIX@/share/man"
+OSNAME="Procursus"
+PREFIX="@MEMO_PREFIX@@MEMO_SUB_PREFIX@"
+MANDIR="${PREFIX}/share/man"
+EXAMPLEDIR="${PREFIX}/share/doc/mandoc"
+MANM_MAN="mandoc_man"           # default is "man"
+MANM_MDOC="mandoc_mdoc"         # default is "mdoc"
+MANM_ROFF="mandoc_roff"         # default is "roff"
+MANM_EQN="mandoc_eqn"           # default is "eqn"
+MANM_TBL="mandoc_tbl"           # default is "tbl"
+BINM_MAN=mman                   # default is "man"
+BINM_APROPOS=mapropos           # default is "apropos"
+BINM_SOELIM=msoelim             # default is "soelim"
+BINM_WHATIS=mwhatis             # default is "whatis"
+BINM_CATMAN=mcatman             # default is "catman"
+BUILD_CATMAN=1
+LN="ln -sf"
+CC="@CC@"
+CFLAGS="@CFLAGS@"
+CPPFLAGS="@CPPFLAGS@"
+LDFLAGS="@LDFLAGS@"
