From 9d8546f2033097fa750e39fffa6594a9c8b44075 Mon Sep 17 00:00:00 2001
From: Hayden <pinseeker75@outlook.com>
Date: Fri, 28 May 2021 18:08:48 -0400
Subject: [PATCH] Support more Apple aarch64 systems

---
 build.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build.rs b/build.rs
index 9cab0eeca..35946ec02 100644
--- a/build.rs
+++ b/build.rs
@@ -80,7 +80,7 @@ fn build_nasm_files() {
   config_file.write(b"	%define PIC 1\n").unwrap();
   config_file.write(b" %define STACK_ALIGNMENT 16\n").unwrap();
   config_file.write(b" %define HAVE_AVX512ICL 1\n").unwrap();
-  if env::var("CARGO_CFG_TARGET_OS").unwrap() == "macos" {
+  if env::var("CARGO_CFG_TARGET_VENDOR").unwrap() == "apple" {
     config_file.write(b" %define PREFIX 1\n").unwrap();
   }
 
@@ -148,7 +148,7 @@ fn build_asm_files() {
 
   let dest_path = Path::new(&out_dir).join("config.h");
   let mut config_file = File::create(&dest_path).unwrap();
-  if env::var("CARGO_CFG_TARGET_OS").unwrap() == "macos" {
+  if env::var("CARGO_CFG_TARGET_VENDOR").unwrap() == "apple" {
     config_file.write(b" #define PREFIX 1\n").unwrap();
   }
   config_file.write(b" #define PRIVATE_PREFIX rav1e_\n").unwrap();
