diff -urN lua5.2/src/Makefile lua5.2/src/Makefile
--- lua5.2/src/Makefile	2013-11-11 06:45:49.000000000 -0500
+++ lua5.2/src/Makefile	2021-03-17 08:36:07.005907136 -0400
@@ -29,6 +29,7 @@
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris

 LUA_A=	liblua.a
+LUA_SO=    liblua$(LUAVER).$(SOVER).dylib
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
 	lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o \
 	ltm.o lundump.o lvm.o lzio.o
@@ -43,7 +44,7 @@
 LUAC_O=	luac.o

 ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O)
-ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T)
+ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T) $(LUA_SO)
 ALL_A= $(LUA_A)

 # Targets start here.
@@ -59,11 +60,14 @@
 	$(AR) $@ $(BASE_O)
 	$(RANLIB) $@

-$(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+$(LUA_SO): $(CORE_O) $(LIB_O)
+	$(CC) -dynamiclib -install_name $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/liblua$(LUAVER).$(SOVER).dylib -o $@ $? $(LDFLAGS)

-$(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+$(LUA_T): $(LUA_O) $(LUA_SO)
+	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_SO) $(LIBS)
+
+$(LUAC_T): $(LUAC_O) $(LUA_SO)
+	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_SO) $(LIBS)

 clean:
 	$(RM) $(ALL_T) $(ALL_O)
@@ -106,7 +107,7 @@
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"

 macosx:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline" CC=cc
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline"

 mingw:
 	$(MAKE) "LUA_A=lua52.dll" "LUA_T=lua.exe" \diff --color -urN lua-5.2.4/src/luaconf.h lua/src/luaconf.h
--- lua-5.2.4/src/luaconf.h	2013-11-21 12:26:16.000000000 -0500
+++ lua/src/luaconf.h	2021-01-14 19:42:32.025287194 -0500
@@ -100,7 +100,7 @@
 #else			/* }{ */
 
 #define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@MEMO_PREFIX@@MEMO_SUB_PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
