diff -ur tapi-1000.10.8/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/CMakeLists.txt
--- tapi-1000.10.8/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -1,6 +1,6 @@
-if(NOT APPLE)
-  message(FATAL_ERROR "Unsupported configuration.")
-endif()
+#if(NOT APPLE)
+#  message(FATAL_ERROR "Unsupported configuration.")
+#endif()
 
 set(TAPI_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(TAPI_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
@@ -38,11 +38,11 @@
 if(DEFINED ENV{RC_SUPPORTED_ARCHS})
   string(REPLACE " " ";" TAPI_SUPPORTED_ARCHS $ENV{RC_SUPPORTED_ARCHS})
 elseif(NOT DEFINED TAPI_SUPPORTED_ARCHS)
-  set(TAPI_SUPPORTED_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64)
+  set(TAPI_SUPPORTED_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64 arm64e)
 endif()
 message(STATUS "Supported Architectures: ${TAPI_SUPPORTED_ARCHS}")
 
-set(KNOWN_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64)
+set(KNOWN_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64 arm64e)
 
 set (CONFIG_FILE ${CMAKE_CURRENT_BINARY_DIR}/include/tapi/Core/ArchitectureConfig.h)
 file(WRITE ${CONFIG_FILE} "#ifndef TAPI_CORE_ARCHITECTURE_CONFIG_H\n")
diff -ur tapi-1000.10.8/include/tapi/Core/Architecture.def apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/Architecture.def
--- tapi-1000.10.8/include/tapi/Core/Architecture.def	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/Architecture.def	2020-01-07 10:20:42.000000000 -0500
@@ -63,10 +63,14 @@
 
 
 ///
-/// ARM64 architectures sorted by cpu sub type id.
+/// ARM64, ARM64E architectures sorted by cpu sub type id.
 ///
 
 #ifdef SUPPORT_ARCH_ARM64
 ARCHINFO(arm64, MachO::CPU_TYPE_ARM64, MachO::CPU_SUBTYPE_ARM64_ALL)
 #endif
 
+
+#ifdef SUPPORT_ARCH_ARM64E
+ARCHINFO(arm64e, MachO::CPU_TYPE_ARM64E, MachO::CPU_SUBTYPE_ARM64E)
+#endif
diff -ur tapi-1000.10.8/include/tapi/Core/FileManager.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/FileManager.h
--- tapi-1000.10.8/include/tapi/Core/FileManager.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/FileManager.h	2020-01-07 10:20:42.000000000 -0500
@@ -42,6 +42,7 @@
 
 /// \brief Basically the clang FileManager with additonal convenience methods
 ///        and a recording stat cache.
+
 class FileManager final : public clang::FileManager {
 public:
   FileManager(const clang::FileSystemOptions &fileSystemOpts,
diff -ur tapi-1000.10.8/include/tapi/Core/HeaderFile.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/HeaderFile.h
--- tapi-1000.10.8/include/tapi/Core/HeaderFile.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/HeaderFile.h	2020-01-07 10:20:42.000000000 -0500
@@ -19,6 +19,7 @@
 #include "tapi/Defines.h"
 #include "llvm/ADT/StringRef.h"
 #include <string>
+#include <vector>
 
 TAPI_NAMESPACE_INTERNAL_BEGIN
 
diff -ur tapi-1000.10.8/include/tapi/Core/Platform.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/Platform.h
--- tapi-1000.10.8/include/tapi/Core/Platform.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Core/Platform.h	2020-01-07 10:20:42.000000000 -0500
@@ -34,6 +34,8 @@
   watchOS,
   watchOSSimulator,
   bridgeOS,
+  iOSMac,
+  zippered
 };
 
 Platform mapToSim(Platform platform, bool wantSim);
diff -ur tapi-1000.10.8/include/tapi/Diagnostics/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/include/tapi/Diagnostics/CMakeLists.txt
--- tapi-1000.10.8/include/tapi/Diagnostics/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Diagnostics/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -1,3 +1,5 @@
+#[[
 tapi_clang_tablegen(DiagnosticTAPIKinds.inc -gen-clang-diags-defs
   SOURCE DiagnosticTAPIKinds.td
   TARGET TAPIDiagnosticTAPI)
+]]
diff -ur tapi-1000.10.8/include/tapi/Driver/DriverOptions.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/Driver/DriverOptions.h
--- tapi-1000.10.8/include/tapi/Driver/DriverOptions.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Driver/DriverOptions.h	2020-01-07 10:20:42.000000000 -0500
@@ -36,7 +36,7 @@
 #define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, PARAM,  \
                HELPTEXT, METAVAR, VALUES)                                      \
   OPT_##ID,
-#include "tapi/Driver/TAPIOptions.inc"
+//#include "tapi/Driver/TAPIOptions.inc"
   LastOption
 #undef OPTION
 };
diff -ur tapi-1000.10.8/include/tapi/Driver/SnapshotFileSystem.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/Driver/SnapshotFileSystem.h
--- tapi-1000.10.8/include/tapi/Driver/SnapshotFileSystem.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Driver/SnapshotFileSystem.h	2020-01-07 10:20:42.000000000 -0500
@@ -11,6 +11,7 @@
 /// \brief Defines the Snapshot Virtual File System.
 ///
 //===----------------------------------------------------------------------===//
+#define TAPI_DRIVER_SNAPSHOT_FILE_SYSTEM_H
 #ifndef TAPI_DRIVER_SNAPSHOT_FILE_SYSTEM_H
 #define TAPI_DRIVER_SNAPSHOT_FILE_SYSTEM_H
 
diff -ur tapi-1000.10.8/include/tapi/Driver/Snapshot.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/Driver/Snapshot.h
--- tapi-1000.10.8/include/tapi/Driver/Snapshot.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/Driver/Snapshot.h	2020-01-07 10:20:42.000000000 -0500
@@ -6,7 +6,7 @@
 // License. See LICENSE.TXT for details.
 //
 //===----------------------------------------------------------------------===//
-
+#define TAPI_DRIVER_SNAPSHOT_H
 #ifndef TAPI_DRIVER_SNAPSHOT_H
 #define TAPI_DRIVER_SNAPSHOT_H
 
diff -ur tapi-1000.10.8/include/tapi/LinkerInterfaceFile.h apple-libtapi-1000.10.8/src/libtapi/include/tapi/LinkerInterfaceFile.h
--- tapi-1000.10.8/include/tapi/LinkerInterfaceFile.h	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/include/tapi/LinkerInterfaceFile.h	2020-01-07 10:20:42.000000000 -0500
@@ -64,6 +64,8 @@
   /// \since 1.2
   bridgeOS = 5,
 
+  iOSMac = 6,
+  zippered = 10
 };
 
 ///
diff -ur tapi-1000.10.8/lib/Core/Platform.cpp apple-libtapi-1000.10.8/src/libtapi/lib/Core/Platform.cpp
--- tapi-1000.10.8/lib/Core/Platform.cpp	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/lib/Core/Platform.cpp	2020-01-07 10:20:42.000000000 -0500
@@ -93,6 +93,10 @@
     return "tvOSSimulator";
   case Platform::bridgeOS:
     return "bridgeOS";
+  case Platform::iOSMac:
+    return "iOSMac";
+  case Platform::zippered:
+    return "zippered";
   }
 }
 
@@ -116,6 +120,10 @@
     return "tvos" + version + "-simulator";
   case Platform::bridgeOS:
     return "bridgeos" + version;
+  case Platform::iOSMac:
+    return "iOSMac" + version;
+  case Platform::zippered:
+    return "zippered" + version;
   }
 }
 
diff -ur tapi-1000.10.8/lib/Core/YAML.cpp apple-libtapi-1000.10.8/src/libtapi/lib/Core/YAML.cpp
--- tapi-1000.10.8/lib/Core/YAML.cpp	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/lib/Core/YAML.cpp	2020-01-07 10:20:42.000000000 -0500
@@ -55,6 +55,9 @@
   io.enumCase(platform, "tvos", Platform::tvOS);
   io.enumCase(platform, "tvos", Platform::tvOSSimulator);
   io.enumCase(platform, "bridgeos", Platform::bridgeOS);
+
+  io.enumCase(platform, "iosmac", Platform::zippered);
+  io.enumCase(platform, "zippered", Platform::zippered);
 }
 
 using TAPI_INTERNAL::Architecture;
diff -ur tapi-1000.10.8/lib/Diagnostics/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/lib/Diagnostics/CMakeLists.txt
--- tapi-1000.10.8/lib/Diagnostics/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/lib/Diagnostics/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -1,3 +1,4 @@
+#[[
 add_tapi_library(tapiDiagnostics
   Diagnostics.cpp
 
@@ -7,3 +8,4 @@
   LINK_LIBS
   tapiCore
   )
+]]
diff -ur tapi-1000.10.8/lib/Driver/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/lib/Driver/CMakeLists.txt
--- tapi-1000.10.8/lib/Driver/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/lib/Driver/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -1,3 +1,4 @@
+#[[
 add_tapi_library(tapiDriver
   API2XPIConverter.cpp
   ArchiveDriver.cpp
@@ -24,3 +25,4 @@
   tapiDiagnostics
   tapiFrontend
   )
+]]
diff -ur tapi-1000.10.8/tools/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/tools/CMakeLists.txt
--- tapi-1000.10.8/tools/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/tools/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -1,4 +1,4 @@
 add_subdirectory(libtapi)
-add_subdirectory(tapi)
-add_subdirectory(tapi-run)
-add_subdirectory(tapi-frontend)
+#add_subdirectory(tapi)
+#add_subdirectory(tapi-run)
+#add_subdirectory(tapi-frontend)
diff -ur tapi-1000.10.8/tools/libtapi/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/tools/libtapi/CMakeLists.txt
--- tapi-1000.10.8/tools/libtapi/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/tools/libtapi/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -20,10 +20,12 @@
   OUTPUT_NAME "tapi"
   )
 
+if(APPLE)
+
 set(DYLIB_VERSION "${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}")
 set_property(TARGET libtapi APPEND_STRING
   PROPERTY
-  LINK_FLAGS " -current_version ${DYLIB_VERSION} -compatibility_version 1 -allowable_client ld"
+  LINK_FLAGS " -current_version ${DYLIB_VERSION} -compatibility_version 1 " #-allowable_client ld"
   )
 
 if(LINKER_SUPPORTS_NO_INITS)
@@ -32,3 +34,6 @@
     LINK_FLAGS " -Wl,-no_inits"
     )
 endif()
+
+endif()
+
diff -ur tapi-1000.10.8/tools/libtapi/libtapi.cpp apple-libtapi-1000.10.8/src/libtapi/tools/libtapi/libtapi.cpp
--- tapi-1000.10.8/tools/libtapi/libtapi.cpp	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/tools/libtapi/libtapi.cpp	2020-01-07 10:20:42.000000000 -0500
@@ -47,12 +47,16 @@
 
 namespace sys {
 
+#ifdef __APPLE__
+
 bool RemoveFileOnSignal(StringRef Filename, std::string *ErrMsg) { abort(); }
 
 void DontRemoveFileOnSignal(StringRef Filename) { abort(); }
 
 void AddSignalHandler(void (*)(void *), void *) { abort(); }
 
+#endif
+
 } // end namespace sys.
 
 } // end namespace llvm.
diff -ur tapi-1000.10.8/tools/libtapi/LinkerInterfaceFile.cpp apple-libtapi-1000.10.8/src/libtapi/tools/libtapi/LinkerInterfaceFile.cpp
--- tapi-1000.10.8/tools/libtapi/LinkerInterfaceFile.cpp	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/tools/libtapi/LinkerInterfaceFile.cpp	2020-01-07 10:20:42.000000000 -0500
@@ -298,6 +298,10 @@
     return Platform::tvOS;
   case tapi::internal::Platform::bridgeOS:
     return Platform::bridgeOS;
+  case tapi::internal::Platform::iOSMac:
+    return Platform::iOSMac;
+  case tapi::internal::Platform::zippered:
+    return Platform::zippered;
   }
 }
 
diff -ur tapi-1000.10.8/tools/tapi/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/tools/tapi/CMakeLists.txt
--- tapi-1000.10.8/tools/tapi/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/tools/tapi/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -15,10 +15,12 @@
   tapi.cpp
   )
 
+
 target_link_libraries(tapi
   tapiDriver
   )
 
+
 add_dependencies(tapi tapi-clang-headers)
 
 install(TARGETS tapi
@@ -28,4 +30,4 @@
 
 add_llvm_install_targets(install-tapi
                          DEPENDS tapi install-tapi-clang-headers
-                         COMPONENT tapi)
+                         COMPONENT tapi)
\ No newline at end of file
diff -ur tapi-1000.10.8/tools/tapi-run/CMakeLists.txt apple-libtapi-1000.10.8/src/libtapi/tools/tapi-run/CMakeLists.txt
--- tapi-1000.10.8/tools/tapi-run/CMakeLists.txt	2019-03-22 16:58:37.000000000 -0400
+++ apple-libtapi-1000.10.8/src/libtapi/tools/tapi-run/CMakeLists.txt	2020-01-07 10:20:42.000000000 -0500
@@ -19,3 +19,4 @@
 add_llvm_install_targets(install-tapi-run
                          DEPENDS tapi-run
                          COMPONENT tapi-run)
+
