--- llvm/lldb/bindings/interfaces.swig	2022-11-12 00:14:17.000000000 +0100
+++ llvm/lldb/bindings/interfaces.swig	2023-02-22 23:47:51.000000000 +0100
@@ -2,7 +2,6 @@
 #define __extension__ /* Undefine GCC keyword to make Swig happy when processing glibc's stdint.h. */
 /* The ISO C99 standard specifies that in C++ implementations limit macros such
    as INT32_MAX should only be defined if __STDC_LIMIT_MACROS is. */
-#define __STDC_LIMIT_MACROS
 %include "stdint.i"
 
 %include "lldb/lldb-defines.h"
