diff -ur llvm/lldb/tools/darwin-debug/darwin-debug.cpp llvm/lldb/tools/darwin-debug/darwin-debug.cpp
--- llvm/lldb/tools/darwin-debug/darwin-debug.cpp	2020-06-08 12:30:56.000000000 -0700
+++ llvm/lldb/tools/darwin-debug/darwin-debug.cpp	2020-06-10 22:04:43.000000000 -0700
@@ -157,6 +157,23 @@
   return pid;
 }
 
+char **environ;
+
+int RunCmd(char *cmd) {
+     pid_t pid;
+     char *argv[] = {strdup("sh"), strdup("-c"), cmd, NULL};
+     int status;
+     status = posix_spawn(&pid, strdup("/bin/sh"), NULL, NULL, argv, environ);
+     if (status == 0) {
+         if (waitpid(pid, &status, 0) == -1) {
+             perror("waitpid");
+         }
+     } else {
+         printf("posix_spawn: %s\n", strerror(status));
+     }
+     return status;
+}
+
 int main(int argc, char *const *argv, char *const *envp, const char **apple) {
 #if defined(DEBUG_LLDB_LAUNCHER)
   const char *program_name = strrchr(apple[0], '/');
@@ -305,7 +322,7 @@
   // We are done with the socket
   close(s);
 
-  system("clear");
+  RunCmd(strdup("clear"));
   printf("Launching: '%s'\n", argv[0]);
   if (working_dir.empty()) {
     char cwd[PATH_MAX];
