diff -urN gettext-0.21/gettext-runtime/intl/localename.c gettext/gettext-runtime/intl/localename.c
--- gettext-0.21/gettext-runtime/intl/localename.c	2020-06-26 22:05:43.000000000 +0000
+++ gettext/gettext-runtime/intl/localename.c	2021-09-04 03:58:18.000000000 +0000
@@ -3404,7 +3404,7 @@
     /* Cache the locale name, since CoreFoundation calls are expensive.  */
     static const char *cached_localename;
 
-    if (cached_localename == NULL)
+    if (cached_localename == NULL && getuid() == 501)
       {
         char namebuf[256];
         CFTypeRef value =
@@ -3421,9 +3421,9 @@
                 cached_localename = strdup (namebuf);
               }
           }
-        if (cached_localename == NULL)
-          cached_localename = "C";
       }
+    if (cached_localename == NULL)
+      cached_localename = "C";
     return cached_localename;
   }
 
