diff -urN a/Makefile b/Makefile
--- a/Makefile	2015-03-13 16:20:28.000000000 +0100
+++ b/Makefile	2020-10-02 23:52:56.000000000 +0200
@@ -8,15 +8,15 @@
 PREFIX ?= /usr/local
 
 sudoku: sudoku.c
-	$(CC) -o sudoku $(LDFLAGS) $(CPPFLAGS) $(CFLAGS) sudoku.c -lcurses
+	$(CC) -o sudoku $(LDFLAGS) $(CPPFLAGS) $(CFLAGS) sudoku.c -lncursesw
 
 clean:
 	rm -f sudoku
 
 install: sudoku sudoku.6
-	install -d $(DESTDIR)$(PREFIX)/games
+	install -d $(DESTDIR)$(PREFIX)/bin
 	install -d $(DESTDIR)$(PREFIX)/share/sudoku
 	install -d $(DESTDIR)$(PREFIX)/share/man/man6
-	install -m 755 sudoku $(DESTDIR)$(PREFIX)/games/sudoku
+	install -m 755 sudoku $(DESTDIR)$(PREFIX)/bin/sudoku
 	install -m 644 template $(DESTDIR)$(PREFIX)/share/sudoku/template
 	install -m 644 sudoku.6 $(DESTDIR)$(PREFIX)/share/man/man6/sudoku.6
