diff --color -urN lua-5.0.3/Makefile lua50/Makefile
--- lua-5.0.3/Makefile	2004-03-11 20:50:55.000000000 -0500
+++ lua50/Makefile	2021-01-14 16:21:54.067793081 -0500
@@ -38,9 +38,9 @@
 
 # shared libraries (for Linux)
 so:
-	ld -o lib/liblua.so.$V -shared src/*.o
-	ld -o lib/liblualib.so.$V -shared src/lib/*.o
-	cd lib; ln -fs liblua.so.$V liblua.so; ln -fs liblualib.so.$V liblualib.so
+	$(CC) $(LDFLAGS) -o lib/liblua50.5.dylib -dynamiclib -install_name $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/liblua50.5.dylib src/*.o
+	$(CC) $(LDFLAGS) -o lib/liblualib50.5.dylib -dynamiclib -install_name $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/liblualib50.5.dylib src/lib/*.o lib/liblua.a
+	cd lib; ln -fs liblua50.5.dylib liblua50.dylib; ln -fs liblualib50.5.dylib liblualib50.dylib; ln -fs liblua50.5.dylib liblua50.5.0.dylib; ln -fs liblualib50.5.dylib liblualib50.5.0.dylib
 
 # binaries using shared libraries
 sobin:
@@ -50,12 +50,12 @@
 
 # install shared libraries
 soinstall:
-	$(INSTALL_EXEC) lib/*.so.* $(INSTALL_LIB)
-	cd $(INSTALL_LIB); ln -fs liblua.so.$V liblua.so; ln -fs liblualib.so.$V liblualib.so
+	$(INSTALL_EXEC) lib/*.dylib $(INSTALL_LIB)
+	cd $(INSTALL_LIB); ln -fs liblua50.5.dylib liblua.dylib; ln -fs liblualib50.5.dylib liblualib50.dylib; ln -fs liblua50.5.dylib liblua50.5.0.dylib; ln -fs liblualib50.5.dylib liblualib50.5.0.dylib
 
 # clean shared libraries
 soclean:
-	rm -f lib/*.so* bin/*
+	rm -f lib/*.dylib bin/*
 
 # echo config parameters
 echo:
