--- a/pfetch	2020-03-11 16:39:16.000000000 -0400
+++ b/pfetch	2021-05-01 11:41:50.000000000 -0400
@@ -208,7 +208,7 @@
             distro="${distro}${WSLENV+ on Windows 10 [WSL2]}"
         ;;
 
-        Darwin*)
+        (Darwin*)
             # Parse the SystemVersion.plist file to grab the macOS
             # version. The file is in the following format:
             #
@@ -231,24 +231,40 @@
                     ;;
                 esac
             done < /System/Library/CoreServices/SystemVersion.plist
+            while IFS='<>' read -r _ _ line _; do
+                case $line in
+                    # Match 'ProductVersion' and read the next line
+                    # directly as it contains the key's value.
+                    ProductName)
+                        IFS='<>' read -r _ _ product_name _
+                        break
+                    ;;
+                esac
+            done < /System/Library/CoreServices/SystemVersion.plist
 
             # Use the ProductVersion to determine which macOS/OS X codename
             # the system has. As far as I'm aware there's no "dynamic" way
             # of grabbing this information.
+            
             case $mac_version in
-                10.4*)  distro='Mac OS X Tiger' ;;
-                10.5*)  distro='Mac OS X Leopard' ;;
-                10.6*)  distro='Mac OS X Snow Leopard' ;;
-                10.7*)  distro='Mac OS X Lion' ;;
-                10.8*)  distro='OS X Mountain Lion' ;;
-                10.9*)  distro='OS X Mavericks' ;;
-                10.10*) distro='OS X Yosemite' ;;
-                10.11*) distro='OS X El Capitan' ;;
-                10.12*) distro='macOS Sierra' ;;
-                10.13*) distro='macOS High Sierra' ;;
-                10.14*) distro='macOS Mojave' ;;
-                10.15*) distro='macOS Catalina' ;;
-                *)      distro='macOS' ;;
+                (10.4*)  distro='Mac OS X Tiger' ;;
+                (10.5*)  distro='Mac OS X Leopard' ;;
+                (10.6*)  distro='Mac OS X Snow Leopard' ;;
+                (10.7*)  distro='Mac OS X Lion' ;;
+                (10.8*)  distro='OS X Mountain Lion' ;;
+                (10.9*)  distro='OS X Mavericks' ;;
+                (10.10*) distro='OS X Yosemite' ;;
+                (10.11*) distro='OS X El Capitan' ;;
+                (10.12*) distro='macOS Sierra' ;;
+                (10.13*) distro='macOS High Sierra' ;;
+                (10.14*) distro='macOS Mojave' ;;
+                (10.15*) distro='macOS Catalina' ;;
+                (*)      distro='macOS' ;;
+                
+            esac
+            case $product_name in
+                (iPhone*)      distro='iOS' ;;
+                
             esac
 
             distro="$distro $mac_version"
@@ -494,6 +510,9 @@
                 # Directories containing packages.
                 has brew       && printf '%s\n' /usr/local/Cellar/*
 
+                # Commands which print packages one per line.
+                has dpkg       && dpkg-query -f '.\n' -W
+
                 # 'port' prints a single line of output to 'stdout'
                 # when no packages are installed and exits with
                 # success causing a false-positive of 1 package
