From 3db3b9494924fce4b24c16f829cd6e78c96bbcd6 Mon Sep 17 00:00:00 2001
From: "Jay Freeman (saurik)" <saurik@saurik.com>
Date: Mon, 14 Jun 2021 00:36:11 -0700
Subject: [PATCH] Fix LDID_NOPLIST (used by Cydia Substrate on iOS).

---
 ldid.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/ldid.cpp b/ldid.cpp
index f5f1bb5..3411e59 100644
--- a/ldid.cpp
+++ b/ldid.cpp
@@ -634,6 +634,7 @@ static std::string der(const std::pair<std::string, std::string> &value) {
     return data.str();
 }
 
+#ifndef LDID_NOPLIST
 static std::string der(plist_t data) {
     switch (const auto type = plist_get_node_type(data)) {
         case PLIST_BOOLEAN: {
@@ -714,6 +715,7 @@ static std::string der(plist_t data) {
         } break;
     }
 }
+#endif
 
 static inline uint16_t Swap_(uint16_t value) {
     return
-- 
1.7.10

