From f9a36c91852165a76a879d655c5c40ea1747cd99 Mon Sep 17 00:00:00 2001
From: sunflsks <22717474+sunflsks@users.noreply.github.com>
Date: Thu, 7 Oct 2021 11:11:42 -0500
Subject: [PATCH] ossl3 compat

---
 ldid.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/ldid.cpp b/ldid.cpp
index b017ebe..31af431 100644
--- a/ldid.cpp
+++ b/ldid.cpp
@@ -43,6 +43,7 @@
 #include <sys/types.h>
 
 #ifndef LDID_NOSMIME
+#include <openssl/provider.h>
 #include <openssl/err.h>
 #include <openssl/pem.h>
 #include <openssl/pkcs7.h>
@@ -3105,6 +3106,8 @@ static void usage(const char *argv0) {
 int main(int argc, char *argv[]) {
 #ifndef LDID_NOSMIME
     OpenSSL_add_all_algorithms();
+    OSSL_PROVIDER *legacy = OSSL_PROVIDER_load(NULL, "legacy");
+    OSSL_PROVIDER *deflt = OSSL_PROVIDER_load(NULL, "default");
 #endif
 
     union {
@@ -3579,6 +3582,11 @@ int main(int argc, char *argv[]) {
         ++filei;
     }
 
+#ifndef LDID_NOSMINE
+    OSSL_PROVIDER_unload(legacy);
+    OSSL_PROVIDER_unload(deflt);
+#endif
+
     return filee;
 }
 #endif
-- 
2.33.0

