From d6ddc694d617087b6be5b938bdd4b9181db43ff1 Mon Sep 17 00:00:00 2001
From: aspen <aspenuwu@protonmail.com>
Date: Mon, 8 Mar 2021 15:30:12 -0500
Subject: [PATCH 2/2] Switch open to iOS-compatible fork

---
 Cargo.lock | 6 +++---
 Cargo.toml | 3 ++-
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 4ab94f5..3a9bad2 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -781,10 +781,10 @@ checksum = "2839e79665f131bdb5782e51f2c6c9599c133c6098982a54c794358bf432529c"
 
 [[package]]
 name = "open"
-version = "1.4.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "7c283bf0114efea9e42f1a60edea9859e8c47528eae09d01df4b29c1e489cc48"
+version = "1.5.1"
+source = "git+https://github.com/aspenluxxxy/open-rs?rev=6e72d687c669c654a5daf6baa8396f4007298f07#6e72d687c669c654a5daf6baa8396f4007298f07"
 dependencies = [
+ "which",
  "winapi",
 ]
 
diff --git a/Cargo.toml b/Cargo.toml
index e43e2d2..7e6f399 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,7 +56,8 @@ pest_derive = "2.1.0"
 regex = "1.4.3"
 os_info = "3.0.1"
 urlencoding = "1.1.1"
-open = "1.4.0"
+# REVERT THIS WHEN https://github.com/Byron/open-rs/pull/28 IS MERGED
+open = { git = "https://github.com/aspenluxxxy/open-rs", rev = "6e72d687c669c654a5daf6baa8396f4007298f07" }
 unicode-width = "0.1.8"
 term_size = "0.3.2"
 quick-xml = "0.20.0"
-- 
2.24.3 (Apple Git-128)

