diff -urN openjdk/make/CompileJavaModules.gmk openjdk/make/CompileJavaModules.gmk
--- openjdk/make/CompileJavaModules.gmk	2020-12-17 10:30:07.000000000 -0500
+++ openjdk/make/CompileJavaModules.gmk	2020-12-29 19:20:52.000000000 -0500
@@ -118,7 +118,7 @@
     .template \
     #
 
-ifeq ($(call isTargetOs, macosx), true)
+ifeq (false, true)
   # exclude all X11 on Mac.
   java.desktop_EXCLUDES += \
       sun/awt/X11 \
@@ -177,14 +177,26 @@
       sun/awt/X11/XWindowChanges.java \
       sun/awt/X11/XdbeSwapInfo.java \
       sun/awt/X11/XmbTextItem.java \
-      sun/awt/X11/XwcTextItem.java
+      sun/awt/X11/XwcTextItem.java \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/java2d/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/java2d/opengl/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/awt/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/lwawt/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/lwawt/macosx/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/font/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/classes/sun/print/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/apple/laf/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/com/apple/eawt/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/com/apple/eawt/event/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/com/apple/eio/*.java) \
+      $(wildcard $(TOPDIR)/src/java.desktop/macosx/com/apple/laf/*.java)
 endif
 
 ifeq ($(call isTargetOs, windows), true)
   java.desktop_EXCLUDES += com/sun/java/swing/plaf/gtk
 endif
 
-ifeq ($(call isTargetOs, windows macosx), false)
+ifeq ($(call isTargetOs, windows), false)
   java.desktop_EXCLUDE_FILES += sun/awt/AWTCharset.java
 endif
 
