diff -urN openjdk/make/autoconf/lib-freetype.m4.orig openjdk/make/autoconf/lib-freetype.m4
--- openjdk/make/autoconf/lib-freetype.m4	2020-12-29 13:28:21.000000000 -0500
+++ openjdk/make/autoconf/lib-freetype.m4	2020-12-29 13:29:19.000000000 -0500
@@ -126,8 +126,7 @@
   fi
 
   if test "x$FREETYPE_TO_USE" = "xsystem" && \
-     (test "x$OPENJDK_TARGET_OS" = "xwindows" || \
-     test "x$OPENJDK_TARGET_OS" = "xmacosx"); then
+     test "x$OPENJDK_TARGET_OS" = "xwindows"; then
     AC_MSG_ERROR([Only bundled freetype can be specified on Mac and Windows])
   fi
 
