diff -urN openjdk/make/autoconf/libraries.m4.orig openjdk/make/autoconf/libraries.m4
--- openjdk/make/autoconf/libraries.m4	2020-12-29 14:15:09.000000000 -0500
+++ openjdk/make/autoconf/libraries.m4	2020-12-29 14:15:33.000000000 -0500
@@ -40,7 +40,7 @@
 AC_DEFUN_ONCE([LIB_DETERMINE_DEPENDENCIES],
 [
   # Check if X11 is needed
-  if test "x$OPENJDK_TARGET_OS" = xwindows || test "x$OPENJDK_TARGET_OS" = xmacosx; then
+  if test "x$OPENJDK_TARGET_OS" = xwindows; then
     # No X11 support on windows or macosx
     NEEDS_LIB_X11=false
   elif test "x$ENABLE_HEADLESS_ONLY" = xtrue; then
@@ -52,7 +52,7 @@
   fi
 
   # Check if fontconfig is needed
-  if test "x$OPENJDK_TARGET_OS" = xwindows || test "x$OPENJDK_TARGET_OS" = xmacosx; then
+  if test "x$OPENJDK_TARGET_OS" = xwindows; then
     # No fontconfig support on windows or macosx
     NEEDS_LIB_FONTCONFIG=false
   else
