diff -ur go/src/crypto/x509/root_omit.go Procursus/build_work/iphoneos-arm64/1500/golang/src/crypto/x509/root_omit.go
--- go/src/crypto/x509/root_omit.go	2022-02-10 02:12:01.000000000 +0800
+++ Procursus/build_work/iphoneos-arm64/1500/golang/src/crypto/x509/root_omit.go	2022-02-19 13:56:36.789293667 +0800
@@ -3,7 +3,7 @@
 // license that can be found in the LICENSE file.
 
 //go:build ((darwin && arm64) || (darwin && amd64 && ios)) && x509omitbundledroots
-// +build darwin,arm64 darwin,amd64,ios
+// +build darwin,arm64 darwin,amd64,ios ios,amd64 ios,arm64
 // +build x509omitbundledroots
 
 // This file provides the loadSystemRoots func when the
