diff -ur go/src/cmd/internal/sys/supported.go Procursus/build_work/iphoneos-arm64/1500/golang/src/cmd/internal/sys/supported.go
--- go/src/cmd/internal/sys/supported.go	2022-02-10 02:12:01.000000000 +0800
+++ Procursus/build_work/iphoneos-arm64/1500/golang/src/cmd/internal/sys/supported.go	2022-02-19 13:47:15.132313810 +0800
@@ -73,7 +75,7 @@
 		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/ppc64le", "linux/s390x",
 			"android/amd64", "android/arm", "android/arm64", "android/386",
 			"freebsd/amd64",
-			"darwin/amd64", "darwin/arm64",
+			"darwin/amd64", "darwin/arm64", "ios/arm64", "ios/amd64",
 			"windows/amd64", "windows/386", "windows/arm64":
 			return true
 		}
@@ -109,7 +111,7 @@
 		switch platform {
 		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/s390x", "linux/ppc64le",
 			"android/amd64", "android/arm", "android/arm64", "android/386",
-			"darwin/amd64", "darwin/arm64",
+			"darwin/amd64", "darwin/arm64", "ios/amd64", "ios/arm64",
 			"freebsd/amd64":
 			return true
 		}
@@ -122,7 +124,7 @@
 
 func InternalLinkPIESupported(goos, goarch string) bool {
 	switch goos + "/" + goarch {
-	case "darwin/amd64", "darwin/arm64",
+	case "darwin/amd64", "darwin/arm64", "ios/amd64", "ios/arm64",
 		"linux/amd64", "linux/arm64",
 		"android/arm64",
 		"windows-amd64", "windows-386", "windows-arm":
