diff -urN a/apt-pkg/contrib/fileutl.cc c/apt-pkg/contrib/fileutl.cc
--- a/apt-pkg/contrib/fileutl.cc	2026-07-31 04:46:13
+++ c/apt-pkg/contrib/fileutl.cc	2026-08-01 15:10:13
@@ -89,6 +89,10 @@
 #include <sys/prctl.h>
 #endif
 
+#if defined(__APPLE__)
+#include <libproc.h>
+#endif
+
 #include <apti18n.h>
 									/*}}}*/
 
@@ -931,6 +935,27 @@
 	 cerr << "FATAL -> Could not set close on exec " << strerror(errno) << endl;
 	 _Exit(100);
       };
+
+#if defined(__APPLE__)
+        int bufsz = 0, count = 0;
+        struct proc_fdinfo *buf;
+
+        bufsz = proc_pidinfo(getpid(), PROC_PIDLISTFDS, 0, NULL, 0); // Gives us a guess of needed size for buffer
+        buf = (struct proc_fdinfo *)calloc(1, bufsz);
+
+        bufsz = proc_pidinfo(getpid(), PROC_PIDLISTFDS, 0, buf, bufsz);
+        count = bufsz / PROC_PIDLISTFD_SIZE;
+
+        for (int i = 0; i < count; i++) {
+            if (buf[i].proc_fdtype == PROX_FDTYPE_VNODE) {
+                int fd = buf[i].proc_fd;
+                if (fd >= 3 && KeepFDs.find(fd) == KeepFDs.end())
+                    fcntl(fd,F_SETFD,FD_CLOEXEC);
+            }
+        }
+        free(buf);
+        return Process;
+#endif
 
 // If we don't have it, make the close_range() expression a failure; avoids #if inside the if/else
 #ifndef CLOSE_RANGE_CLOEXEC
