diff --git a/apt-pkg/contrib/fileutl.h b/apt-pkg/contrib/fileutl.h
--- a/apt-pkg/contrib/fileutl.h
+++ b/apt-pkg/contrib/fileutl.h
@@ -30,6 +30,15 @@
 
 #include <zlib.h>
 
+/* wrapper functions to make iOS shell scripts function correctly */
+#if defined(__APPLE__)
+#  include <TargetConditionals.h>
+#  if TARGET_OS_IPHONE
+#    define ENABLE_IOSEXEC
+#    include <libiosexec.h>
+#  endif
+#endif
+
 /* Define this for python-apt */
 #define APT_HAS_GZIP 1
 
