From 25dc97e5ca41ac804b821fc6c0a51e1377861dfd Mon Sep 17 00:00:00 2001
From: Hayden <me@diatr.us>
Date: Wed, 19 Aug 2020 00:24:16 -0400
Subject: [PATCH] Fix upstream bug

---
 apt-pkg/acquire-item.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apt-pkg/acquire-item.cc b/apt-pkg/acquire-item.cc
index 9b13a2483..faeec1ed4 100644
--- a/apt-pkg/acquire-item.cc
+++ b/apt-pkg/acquire-item.cc
@@ -3481,7 +3481,7 @@ pkgAcqArchive::pkgAcqArchive(pkgAcquire *const Owner, pkgSourceList *const Sourc
    if (stat(FinalFile.c_str(), &Buf) == 0)
    {
       // Make sure the size matches
-      if ((unsigned long long)Buf.st_size == Version->Size)
+      if ((unsigned long long)Buf.st_size == Version->Size || (Buf.st_size != 0 && Version->Size == 0))
       {
 	 Complete = true;
 	 Local = true;
