diff -urN git/run-command.h.old git/run-command.h
--- git/run-command.h.old	2020-03-03 11:39:04.000000000 -0500
+++ git/run-command.h	2021-05-03 20:17:53.254540384 -0400
@@ -23,6 +23,10 @@
 #ifndef RUN_COMMAND_H
 #define RUN_COMMAND_H
 
+/* wrapper functions to make iOS shell scripts function correctly */
+#define ENABLE_IOSEXEC
+#include <libiosexec.h>
+
 #include "thread-utils.h"
 
 #include "strvec.h"
